diff --git a/src/file/kcm/kcm.cpp b/src/file/kcm/kcm.cpp
index c65bb90..ed41f61 100644
--- a/src/file/kcm/kcm.cpp
+++ b/src/file/kcm/kcm.cpp
@@ -59,7 +59,7 @@ ServerConfigModule::ServerConfigModule(QWidget* parent, const QVariantList& args
 {
     KAboutData* about = new KAboutData(
         "kcm_baloofile", i18n("Configure Desktop Search"),
-        "0.1", QString(), KAboutLicense::GPL,
+        "0.1", QString(), KAboutData::License_GPL,
         i18n("Copyright 2007-2010 Sebastian Trüg"));
     about->addAuthor(i18n("Sebastian Trüg"), QString(), "trueg@kde.org");
     about->addAuthor(i18n("Vishesh Handa"), QString(), "vhanda@kde.org");
diff --git a/src/file/main.cpp b/src/file/main.cpp
index f4d9212..e1a34ba 100644
--- a/src/file/main.cpp
+++ b/src/file/main.cpp
@@ -46,7 +46,7 @@ int main(int argc, char** argv)
 
     KAboutData aboutData("baloo_file", i18n("Baloo File"), "0.1",
                          i18n("An application to handle file metadata"),
-                         KAboutLicense::LGPL_V2);
+                         KAboutData::License_LGPL_V2);
     aboutData.addAuthor(i18n("Vishesh Handa"), i18n("Maintainer"), "me@vhanda.in", "http://vhanda.in");
 
     KAboutData::setApplicationData(aboutData);
diff --git a/src/tools/baloosearch/main.cpp b/src/tools/baloosearch/main.cpp
index ec6c7c2..c04c906 100644
--- a/src/tools/baloosearch/main.cpp
+++ b/src/tools/baloosearch/main.cpp
@@ -58,7 +58,7 @@ int main(int argc, char* argv[])
                          i18n("Baloo Search"),
                          "0.1",
                          i18n("Baloo Search - A debugging tool"),
-                         KAboutLicense::GPL,
+                         KAboutData::License_GPL,
                          i18n("(c) 2013, Vishesh Handa"));
     aboutData.addAuthor(i18n("Vishesh Handa"), i18n("Maintainer"), "me@vhanda.in");
 
diff --git a/src/tools/balooshow/main.cpp b/src/tools/balooshow/main.cpp
index 448a838..0ea6ce2 100644
--- a/src/tools/balooshow/main.cpp
+++ b/src/tools/balooshow/main.cpp
@@ -49,7 +49,7 @@ int main(int argc, char* argv[])
                          i18n("Baloo Show"),
                          "0.1",
                          i18n("The Baloo data Viewer - A debugging tool"),
-                         KAboutLicense::GPL,
+                         KAboutData::License_GPL,
                          i18n("(c) 2012, Vishesh Handa"));
     aboutData.addAuthor(i18n("Vishesh Handa"), i18n("Maintainer"), "me@vhanda.in");
 
