diff --git a/src/ConfigureChecks.cmake b/src/ConfigureChecks.cmake
index f3dc6a6..c3bdc10 100644
--- a/src/ConfigureChecks.cmake
+++ b/src/ConfigureChecks.cmake
@@ -142,13 +142,13 @@ elseif(KAUTH_BACKEND_NAME STREQUAL "POLKITQT")
 elseif(KAUTH_BACKEND_NAME STREQUAL "POLKITQT-1")
     message(STATUS "Building PolkitQt-1 KAuth backend")
 
-    include_directories(${POLKITQT-1_INCLUDE_DIR})
+    include_directories("${POLKITQT-1_INSTALL_DIR}/${POLKITQT-1_INCLUDE_DIR}")
 
     set(KAUTH_BACKEND_SRCS
         backends/polkit-1/Polkit1Backend.cpp
     )
 
-    set(KAUTH_BACKEND_LIBS ${POLKITQT-1_CORE_LIBRARY} Qt5::DBus Qt5::Widgets)
+    set(KAUTH_BACKEND_LIBS "${POLKITQT-1_INSTALL_DIR}/${POLKITQT-1_CORE_LIBRARY}" Qt5::DBus Qt5::Widgets)
 
     # POLKITQT-1_POLICY_FILES_INSTALL_DIR has an absolute pathname, fix that.
     string(REPLACE ${POLKITQT-1_INSTALL_DIR}
