diff --git a/src/ConfigureChecks.cmake b/src/ConfigureChecks.cmake
index f3dc6a6..4aee977 100644
--- a/src/ConfigureChecks.cmake
+++ b/src/ConfigureChecks.cmake
@@ -142,7 +142,7 @@ elseif(KAUTH_BACKEND_NAME STREQUAL "POLKITQT")
 elseif(KAUTH_BACKEND_NAME STREQUAL "POLKITQT-1")
     message(STATUS "Building PolkitQt-1 KAuth backend")
 
-    include_directories(${POLKITQT-1_INCLUDE_DIR})
+    include_directories("${POLKITQT-1_INSTALL_DIR}/${POLKITQT-1_INCLUDE_DIR}")
 
     set(KAUTH_BACKEND_SRCS
         backends/polkit-1/Polkit1Backend.cpp
