diff --git a/CMakeLists.txt b/CMakeLists.txt
index e5ca7e4..700a126 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,7 +42,7 @@ set_package_properties(LibXml2 PROPERTIES
                       )
 
 
-find_package(DocBookXML4 "4.2")
+find_package(DocBookXML4 "4.5")
 
 set_package_properties(DocBookXML4 PROPERTIES
                        TYPE REQUIRED
diff --git a/KF5DocToolsMacros.cmake b/KF5DocToolsMacros.cmake
index 6567b67..7647784 100644
--- a/KF5DocToolsMacros.cmake
+++ b/KF5DocToolsMacros.cmake
@@ -35,6 +35,11 @@ if(KDOCTOOLS_SERIALIZE_TOOL)
     set(KDOCTOOLS_MEINPROC_EXECUTABLE ${KDOCTOOLS_SERIALIZE_TOOL} ${KDOCTOOLS_MEINPROC_EXECUTABLE})
 endif(KDOCTOOLS_SERIALIZE_TOOL)
 
+macro(_SUGGEST_TARGET_NAME _out)
+    string(REPLACE "${CMAKE_SOURCE_DIR}/" "" ${_out} "${CMAKE_CURRENT_SOURCE_DIR}")
+    string(REGEX REPLACE "[^0-9a-zA-Z]" "-" ${_out} "${${_out}}")
+endmacro()
+
 macro (KDOCTOOLS_CREATE_HANDBOOK _docbook)
    get_filename_component(_input ${_docbook} ABSOLUTE)
    set(_doc ${CMAKE_CURRENT_BINARY_DIR}/index.cache.bz2)
@@ -60,7 +65,8 @@ macro (KDOCTOOLS_CREATE_HANDBOOK _docbook)
       DEPENDS ${_docs} ${_ssheet}
       WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
    )
-   get_filename_component(_targ ${CMAKE_CURRENT_SOURCE_DIR} NAME)
+
+   _suggest_target_name(_targ)
    set(_targ "${_targ}-handbook")
    add_custom_target(${_targ} ALL DEPENDS ${_doc})
 
@@ -135,7 +141,7 @@ macro (KDOCTOOLS_CREATE_MANPAGE _docbook _section)
       COMMAND ${KDOCTOOLS_MEINPROC_EXECUTABLE} --stylesheet ${_ssheet} --check ${_bootstrapOption} ${_input}
       DEPENDS ${_input} ${_ssheet}
    )
-   get_filename_component(_targ ${CMAKE_CURRENT_SOURCE_DIR} NAME)
+   _suggest_target_name(_targ)
    set(_targ "${_targ}-manpage-${_base}")
    add_custom_target(${_targ} ALL DEPENDS "${_outdoc}")
 
diff --git a/cmake/FindDocBookXML4.cmake b/cmake/FindDocBookXML4.cmake
index eb4bfd8..415745f 100644
--- a/cmake/FindDocBookXML4.cmake
+++ b/cmake/FindDocBookXML4.cmake
@@ -1,5 +1,5 @@
 # Try to find DocBook XML 4.x DTD.
-# By default it will find version 4.2. A different version can be specified
+# By default it will find version 4.5. A different version can be specified
 # as parameter for find_package().
 # Once done, it will define:
 #
@@ -16,7 +16,7 @@
 
 
 if (NOT DocBookXML_FIND_VERSION)
-     set(DocBookXML_FIND_VERSION "4.2")
+     set(DocBookXML_FIND_VERSION "4.5")
 endif ()
 
 set (DocBookXML4_DTD_VERSION ${DocBookXML_FIND_VERSION}
@@ -70,12 +70,4 @@ find_package_handle_standard_args (DocBookXML4
     REQUIRED_VARS DocBookXML4_DTD_DIR DocBookXML4_DTD_VERSION
     FOUND_VAR DocBookXML4_FOUND)
 
-#maintain backwards compatibility
-# legacy version
-locate_version ("4.2" DOCBOOKXML_CURRENTDTD_DIR)
-if (DOCBOOKXML_CURRENTDTD_DIR)
-    set(DOCBOOKXML_FOUND "TRUE")
-    set(DOCBOOKXML_CURRENTDTD_VERSION "4.2")
-endif ()
-
 mark_as_advanced (DocBookXML4_DTD_DIR DocBookXML4_DTD_VERSION)
diff --git a/docs/checkXML5/man-checkXML5.1.docbook b/docs/checkXML5/man-checkXML5.1.docbook
index 68509b9..15854b5 100644
--- a/docs/checkXML5/man-checkXML5.1.docbook
+++ b/docs/checkXML5/man-checkXML5.1.docbook
@@ -1,5 +1,5 @@
 <?xml version="1.0" ?>
-<!DOCTYPE refentry PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE refentry PUBLIC "-//KDE//DTD DocBook XML V4.5-Based Variant V1.1//EN" "dtd/kdedbx45.dtd" [
 <!ENTITY % English "INCLUDE">
 ]>
 
diff --git a/docs/kf5options/man-kf5options.7.docbook b/docs/kf5options/man-kf5options.7.docbook
index cb7973d..f1c4783 100644
--- a/docs/kf5options/man-kf5options.7.docbook
+++ b/docs/kf5options/man-kf5options.7.docbook
@@ -1,5 +1,5 @@
 <?xml version="1.0" ?>
-<!DOCTYPE refentry PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE refentry PUBLIC "-//KDE//DTD DocBook XML V4.5-Based Variant V1.1//EN" "dtd/kdedbx45.dtd" [
 <!ENTITY % English "INCLUDE">
 ]>
 
diff --git a/docs/meinproc5/man-meinproc5.8.docbook b/docs/meinproc5/man-meinproc5.8.docbook
index 77799b4..7b4c535 100644
--- a/docs/meinproc5/man-meinproc5.8.docbook
+++ b/docs/meinproc5/man-meinproc5.8.docbook
@@ -1,5 +1,5 @@
 <?xml version="1.0" ?>
-<!DOCTYPE refentry PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE refentry PUBLIC "-//KDE//DTD DocBook XML V4.5-Based Variant V1.1//EN" "dtd/kdedbx45.dtd" [
   <!ENTITY % English "INCLUDE"><!-- change language only here -->
 ]>
 
diff --git a/docs/qt5options/man-qt5options.7.docbook b/docs/qt5options/man-qt5options.7.docbook
index 7afbf07..25e2994 100644
--- a/docs/qt5options/man-qt5options.7.docbook
+++ b/docs/qt5options/man-qt5options.7.docbook
@@ -1,5 +1,5 @@
 <?xml version="1.0" ?>
-<!DOCTYPE refentry PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE refentry PUBLIC "-//KDE//DTD DocBook XML V4.5-Based Variant V1.1//EN" "dtd/kdedbx45.dtd" [
 <!ENTITY % English "INCLUDE">
 ]>
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 0329bd3..61e48f3 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,6 +1,6 @@
 include_directories( ${LIBXML2_INCLUDE_DIR} ${LIBXSLT_INCLUDE_DIR} )
 
-set (_custom_dtd_kdex "customization/dtd/kdex.dtd")
+set (_custom_dtd_kdex "customization/dtd/kdedbx45.dtd")
 configure_file(${_custom_dtd_kdex}.cmake ${CMAKE_CURRENT_BINARY_DIR}/${_custom_dtd_kdex} )
 # WARNING: this is due to severe limitations on the windows platform to keep the dtd relocateable
 # see the .xsl.cmake files for a more verbose explanation below
@@ -8,12 +8,12 @@ if(NOT WIN32)
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${_custom_dtd_kdex}
             DESTINATION ${DATA_INSTALL_DIR}/kdoctools5/customization/dtd )
 else()
+    get_filename_component(NORMALIZED_DTD_INSTALL_PATH "${CMAKE_INSTALL_PREFIX}/${DATA_INSTALL_DIR}/kdoctools5/customization/dtd" ABSOLUTE)
     # file(RELATIVE_PATH ...) returns wrong values for paths containing /../ (it doesn't normalize the path)
     # To work around this we make sure the variable passed has all /../ elements
     # removed using get_filename_component(... ABSOLUTE)
     install(CODE
-        "get_filename_component(NORMALIZED_INSTALL_PATH \"${CMAKE_INSTALL_PREFIX}/${DATA_INSTALL_DIR}/kdoctools5/customization/dtd\" ABSOLUTE)
-        file(RELATIVE_PATH DocBookXML4_DTD_DIR \"${NORMALIZED_INSTALL_PATH}\" \"${DocBookXML4_DTD_DIR}\")
+        "file(RELATIVE_PATH DocBookXML4_DTD_DIR \"${NORMALIZED_DTD_INSTALL_PATH}\" \"${DocBookXML4_DTD_DIR}\")
         configure_file(${CMAKE_CURRENT_SOURCE_DIR}/${_custom_dtd_kdex}.cmake ${CMAKE_CURRENT_BINARY_DIR}/${_custom_dtd_kdex}.install)"
     )
     get_filename_component(_custom_dtd_kdex_name ${_custom_dtd_kdex} NAME)
@@ -24,7 +24,10 @@ endif()
 add_library(KF5XsltKde STATIC xslt.cpp xslt_kde.cpp)
 add_library(KF5::XsltKde ALIAS KF5XsltKde)
 target_link_libraries(KF5XsltKde Qt5::Core KF5::Archive)
-set_target_properties(KF5XsltKde PROPERTIES EXPORT_NAME XsltKde)
+set_target_properties(KF5XsltKde PROPERTIES
+    POSITION_INDEPENDENT_CODE TRUE
+    EXPORT_NAME XsltKde
+)
 
 target_include_directories(KF5XsltKde INTERFACE "$<INSTALL_INTERFACE:${INCLUDE_INSTALL_DIR}/XsltKde>")
 
diff --git a/src/customization/catalog.xml b/src/customization/catalog.xml
index 229ae70..31e01eb 100644
--- a/src/customization/catalog.xml
+++ b/src/customization/catalog.xml
@@ -31,10 +31,10 @@ SGMLDECL "xml.dcl"
 
   <!-- DocBook -->
 
-  <!-- DocBook 4.2 KDE Variant V1.1 -->
+  <!-- DocBook 4.5 KDE Variant V1.1 -->
 
-  <public publicId="-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN"
-          uri="dtd/kdex.dtd"/>
+  <public publicId="-//KDE//DTD DocBook XML V4.5-Based Variant V1.1//EN"
+          uri="dtd/kdedbx45.dtd"/>
   <public publicId="-//KDE//ELEMENTS DocBook XML Hierarchy Redeclarations 2 V1.0//EN"
           uri="dtd/rdbhier2.elements"/>
   <public publicId="-//KDE//ELEMENTS DocBook XML Pool Redeclarations V1.1//EN"
diff --git a/src/customization/dtd/kdedbx45.dtd.cmake b/src/customization/dtd/kdedbx45.dtd.cmake
new file mode 100644
index 0000000..0712e24
--- /dev/null
+++ b/src/customization/dtd/kdedbx45.dtd.cmake
@@ -0,0 +1,129 @@
+<!--
+    Document Type Definition for DocBook XML as used in the KDE documentation
+    
+    Copyright (C) 2002 Frederik Fouvry
+     
+    This program is free software; you can redistribute it and/or
+    modify it under the terms of the GNU General Public License
+    as published by the Free Software Foundation; either version 2
+    of the License, or (at your option) any later version.
+     
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+     
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+    
+    Send suggestions, comments, etc. to the KDE docbook list 
+    <kde-docbook@kde.org>.
+
+
+    USAGE
+
+    Refer to this DTD as
+
+      "-//KDE//DTD DocBook XML V4.5-Based Variant V1.1//EN"
+
+    For instance:
+
+      <!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.5-Based Variant V1.1//EN"
+                     "customization/dtd/kdedbx45.dtd">
+
+    and use catalog.xml or docbook.cat as the catalog file.
+    Note: the system identifier is interpreted relative to the place
+    where it is defined, unless otherwise defined by the application you use.
+    If you have permanent internet access, you can use any URLs (and
+    you don't have to install docbook).
+
+    After a version number change, plenty of problems may occur because
+    of non-compatibilities (or DTDs not found, etc).  That doesn't mean your
+    document is invalid; it just means that the reference DTD has changed 
+    (new documentation and documentation updates).  Check out the previous
+    version and point to the appropriate catalog or DTD and things should
+    work again (=still ;-).
+-->
+
+
+<!-- ============================================================= -->
+<!--	      Modifications to the original DocBook DTD		   -->
+<!-- ============================================================= -->
+<!-- Declarations in this part should only *restrict* the DocBook 
+     DTD                                                           -->
+
+<!-- copied from docbookx.dtd: these values are for 
+     %xml.features; = "INCLUDE"
+     Files that define elements need them (currently only 
+     modifications.elements) -->
+<!ENTITY % ho "">
+<!ENTITY % ubiq.inclusion "">
+<!ENTITY % beginpage.exclusion "">
+
+<!-- Modifications in pool
+     %rdbpool; is called in the DocBook DTD
+     NEEDS TO BE CHECKED ON CHANGES
+ -->
+<!ENTITY % rdbpool	    PUBLIC
+  "-//KDE//ELEMENTS DocBook XML Pool Redeclarations V1.1//EN"	     
+  "rdbpool.elements"                                                 >
+<!ENTITY % dbpool.redecl.module "INCLUDE"			     >
+
+<!-- Modifications in hierarchy
+     %rdbhier2; is called in the DocBook DTD
+     NEEDS TO BE CHECKED ON CHANGES
+ -->
+<!ENTITY % rdbhier2	    PUBLIC
+  "-//KDE//ELEMENTS DocBook XML Hierarchy Redeclarations 2 V1.0//EN"     
+  "rdbhier2.elements"                                                >
+<!ENTITY % dbhier.redecl2.module "INCLUDE"			     >
+
+<!-- For Java-style refentries	   
+     Based on DocBook 4.0.                                         -->
+<!--ENTITY % jrefentry      PUBLIC
+  "-//Norman Walsh//DTD JRefEntry V1.0//EN"                          >
+%jrefentry; -->
+
+<!-- Temporary solution -->
+<!ENTITY copy  "(c)"> <!--=copyright sign-->
+<!ENTITY trade "[tm]"> <!--=trade mark sign-->
+<!ENTITY reg   "(R)"> <!--/circledR =registered sign-->
+
+
+<!-- ============================================================= -->
+<!--			 Original DocBook DTD			   -->
+<!-- ============================================================= -->
+
+<!ENTITY % DocBookDTD	    PUBLIC
+  "-//OASIS//DTD DocBook XML V4.5//EN"
+  "@DocBookXML4_DTD_DIR@/docbookx.dtd"         >
+%DocBookDTD;
+
+
+<!-- ============================================================= -->
+<!--          Modifications to the original DocBook DTD            -->
+<!-- ============================================================= -->
+<!-- ONLY to supply declarations that were switched off in the 
+     first "Modifications" section -->
+
+<!ENTITY % kde.modifications PUBLIC
+  "-//KDE//ELEMENTS DocBook XML Modifications V1.1//EN"
+  "modifications.elements"                                           >
+%kde.modifications;
+
+
+<!-- ============================================================= -->
+<!--		    Extensions to the DocBook DTD		   -->
+<!-- ============================================================= -->
+<!-- Is empty, except for what could go in dbgenent.mod            -->
+
+<!ENTITY % kde.general.entities	PUBLIC
+  "-//KDE//ENTITIES DocBook XML General Entity Declarations V1.2//EN"    
+  "../entities/general.entities"                                     >
+%kde.general.entities;
+
+<!ENTITY % kde.i18n.entities PUBLIC
+  "-//KDE//ENTITIES DocBook XML Localisation Entity Declarations V1.0//EN"
+  "../entities/l10n.entities"                                        >
+%kde.i18n.entities;
diff --git a/src/customization/dtd/kdex.dtd.cmake b/src/customization/dtd/kdex.dtd.cmake
deleted file mode 100644
index c2f7b2c..0000000
--- a/src/customization/dtd/kdex.dtd.cmake
+++ /dev/null
@@ -1,129 +0,0 @@
-<!--
-    Document Type Definition for DocBook XML as used in the KDE documentation
-    
-    Copyright (C) 2002 Frederik Fouvry
-     
-    This program is free software; you can redistribute it and/or
-    modify it under the terms of the GNU General Public License
-    as published by the Free Software Foundation; either version 2
-    of the License, or (at your option) any later version.
-     
-    This program is distributed in the hope that it will be useful,
-    but WITHOUT ANY WARRANTY; without even the implied warranty of
-    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-    GNU General Public License for more details.
-     
-    You should have received a copy of the GNU General Public License
-    along with this program; if not, write to the Free Software
-    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
-    
-    Send suggestions, comments, etc. to the KDE docbook list 
-    <kde-docbook@kde.org>.
-
-
-    USAGE
-
-    Refer to this DTD as
-
-      "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN"
-
-    For instance:
-
-      <!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN"
-                     "customization/dtd/kdex.dtd">
-
-    and use catalog.xml or docbook.cat as the catalog file.
-    Note: the system identifier is interpreted relative to the place
-    where it is defined, unless otherwise defined by the application you use.
-    If you have permanent internet access, you can use any URLs (and
-    you don't have to install docbook).
-
-    After a version number change, plenty of problems may occur because
-    of non-compatibilities (or DTDs not found, etc).  That doesn't mean your
-    document is invalid; it just means that the reference DTD has changed 
-    (new documentation and documentation updates).  Check out the previous
-    version and point to the appropriate catalog or DTD and things should
-    work again (=still ;-).
--->
-
-
-<!-- ============================================================= -->
-<!--	      Modifications to the original DocBook DTD		   -->
-<!-- ============================================================= -->
-<!-- Declarations in this part should only *restrict* the DocBook 
-     DTD                                                           -->
-
-<!-- copied from docbookx.dtd: these values are for 
-     %xml.features; = "INCLUDE"
-     Files that define elements need them (currently only 
-     modifications.elements) -->
-<!ENTITY % ho "">
-<!ENTITY % ubiq.inclusion "">
-<!ENTITY % beginpage.exclusion "">
-
-<!-- Modifications in pool
-     %rdbpool; is called in the DocBook DTD
-     NEEDS TO BE CHECKED ON CHANGES
- -->
-<!ENTITY % rdbpool	    PUBLIC
-  "-//KDE//ELEMENTS DocBook XML Pool Redeclarations V1.1//EN"	     
-  "rdbpool.elements"                                                 >
-<!ENTITY % dbpool.redecl.module "INCLUDE"			     >
-
-<!-- Modifications in hierarchy
-     %rdbhier2; is called in the DocBook DTD
-     NEEDS TO BE CHECKED ON CHANGES
- -->
-<!ENTITY % rdbhier2	    PUBLIC
-  "-//KDE//ELEMENTS DocBook XML Hierarchy Redeclarations 2 V1.0//EN"     
-  "rdbhier2.elements"                                                >
-<!ENTITY % dbhier.redecl2.module "INCLUDE"			     >
-
-<!-- For Java-style refentries	   
-     Based on DocBook 4.0.                                         -->
-<!--ENTITY % jrefentry      PUBLIC
-  "-//Norman Walsh//DTD JRefEntry V1.0//EN"                          >
-%jrefentry; -->
-
-<!-- Temporary solution -->
-<!ENTITY copy  "(c)"> <!--=copyright sign-->
-<!ENTITY trade "[tm]"> <!--=trade mark sign-->
-<!ENTITY reg   "(R)"> <!--/circledR =registered sign-->
-
-
-<!-- ============================================================= -->
-<!--			 Original DocBook DTD			   -->
-<!-- ============================================================= -->
-
-<!ENTITY % DocBookDTD	    PUBLIC
-  "-//OASIS//DTD DocBook XML V4.2//EN"
-  "@DocBookXML4_DTD_DIR@/docbookx.dtd"         >
-%DocBookDTD;
-
-
-<!-- ============================================================= -->
-<!--          Modifications to the original DocBook DTD            -->
-<!-- ============================================================= -->
-<!-- ONLY to supply declarations that were switched off in the 
-     first "Modifications" section -->
-
-<!ENTITY % kde.modifications PUBLIC
-  "-//KDE//ELEMENTS DocBook XML Modifications V1.1//EN"
-  "modifications.elements"                                           >
-%kde.modifications;
-
-
-<!-- ============================================================= -->
-<!--		    Extensions to the DocBook DTD		   -->
-<!-- ============================================================= -->
-<!-- Is empty, except for what could go in dbgenent.mod            -->
-
-<!ENTITY % kde.general.entities	PUBLIC
-  "-//KDE//ENTITIES DocBook XML General Entity Declarations V1.2//EN"    
-  "../entities/general.entities"                                     >
-%kde.general.entities;
-
-<!ENTITY % kde.i18n.entities PUBLIC
-  "-//KDE//ENTITIES DocBook XML Localisation Entity Declarations V1.0//EN"
-  "../entities/l10n.entities"                                        >
-%kde.i18n.entities;
diff --git a/src/man-template.docbook b/src/man-template.docbook
index bdc88c7..6c2e995 100644
--- a/src/man-template.docbook
+++ b/src/man-template.docbook
@@ -1,5 +1,5 @@
 <?xml version="1.0" ?>
-<!DOCTYPE refentry PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE refentry PUBLIC "-//KDE//DTD DocBook XML V4.5-Based Variant V1.1//EN" "dtd/kdedbx45.dtd" [
 <!ENTITY % English "INCLUDE">
 ]>
 
diff --git a/src/meinproc.cpp b/src/meinproc.cpp
index 95adcea..22150ab 100644
--- a/src/meinproc.cpp
+++ b/src/meinproc.cpp
@@ -167,16 +167,6 @@ int main(int argc, char **argv)
     xmlLoadExtDtdDefaultValue = 1;
 
     QVector<const char *> params;
-#ifndef Q_OS_WIN
-    // libxslt parses the path given to outputFile as XPath expression which fails
-    // see libxslt/xsltEvalUserParams
-    // this parameter is used only by share/kdoctools5/docbook/xsl/html/math.xsl
-    // and is not supported on windows yet
-    if (parser.isSet(QStringLiteral("output"))) {
-        params.append(qstrdup("outputFile"));
-        params.append(qstrdup(parser.value(QStringLiteral("output")).toLocal8Bit().constData()));
-    }
-#endif
     {
         const QStringList paramList = parser.values(QStringLiteral("param"));
         QStringList::ConstIterator it = paramList.constBegin();
diff --git a/src/template.docbook b/src/template.docbook
index 08762e5..3b5dc8a 100644
--- a/src/template.docbook
+++ b/src/template.docbook
@@ -1,5 +1,5 @@
 <?xml version="1.0" ?>
-<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.2-Based Variant V1.1//EN" "dtd/kdex.dtd" [
+<!DOCTYPE book PUBLIC "-//KDE//DTD DocBook XML V4.5-Based Variant V1.1//EN" "dtd/kdedbx45.dtd" [
   <!-- Define an entity for your application if it is not part of KDE
        Git -->
   <!ENTITY kmyapplication "<application>KMyApp</application>">
diff --git a/src/xslt.cpp b/src/xslt.cpp
index 280f51b..db67599 100644
--- a/src/xslt.cpp
+++ b/src/xslt.cpp
@@ -145,7 +145,7 @@ QString transform(const QString &pat, const QString &tss,
         defaultEntityLoader = xmlGetExternalEntityLoader();
         xmlSetExternalEntityLoader(xsltprocExternalEntityLoader);
 
-        replaceURLList[QLatin1String("http://www.oasis-open.org/docbook/xml/4.2")] = QString("file:///%1").arg(DOCBOOK_XML_CURRDTD);
+        replaceURLList[QLatin1String("http://www.oasis-open.org/docbook/xml/4.5")] = QString("file:///%1").arg(DOCBOOK_XML_CURRDTD);
     }
 #endif
 
@@ -373,48 +373,61 @@ void setupStandardDirs(const QString &srcdir)
 
 QString locateFileInDtdResource(const QString &file, const QStandardPaths::LocateOptions option)
 {
+    const QStringList lst = locateFilesInDtdResource(file, option);
+    return lst.isEmpty() ? QString() : lst.first();
+}
+
+QStringList locateFilesInDtdResource(const QString &file, const QStandardPaths::LocateOptions option)
+{
     QFileInfo info(file);
     if (info.exists() && info.isAbsolute()) {
-        return file;
+        return QStringList() << file;
     }
 
     const QString srcdir = s_dtdDirs()->srcdir;
     if (!srcdir.isEmpty()) {
         const QString test = srcdir + QLatin1Char('/') + file;
         if (QFile::exists(test)) {
-            return test;
+            return QStringList() << test;
         }
-        qDebug() << "Could not locate file:" << file;
-        return QString();
+        qDebug() << "Could not locate file" << file << "in" << srcdir;
+        return QStringList();
     }
-    const QString result = QStandardPaths::locate(QStandardPaths::GenericDataLocation,
+    // Using locateAll() is necessary to be able to find all catalogs when
+    // running in environments where every repository is installed in its own
+    // prefix.
+    // This is the case on build.kde.org where kde4support installs catalogs
+    // in a different prefix than kdoctools.
+    const QStringList result = QStandardPaths::locateAll(QStandardPaths::GenericDataLocation,
             QStringLiteral("kdoctools5/") + file, option);
     if (result.isEmpty()) {
-        qDebug() << "Could not locate file:" << file;
+        qDebug() << "Could not locate file" << file << "in standard dirs";
     }
     return result;
 }
 
 QStringList getKDocToolsCatalogs()
 {
-    // find all catalogs as catalog*.xml, and add them to the list,
-    // starting from catalog.xml (the main one).
-    const QString customizationDirName = locateFileInDtdResource(QStringLiteral("customization"),
-                                                                 QStandardPaths::LocateDirectory);
-    if (customizationDirName.isEmpty()) {
+    // Find all catalogs as catalog*.xml, and add them to the list, starting
+    // from catalog.xml (the main one).
+    QStringList dirNames = locateFilesInDtdResource(QStringLiteral("customization"),
+                                                    QStandardPaths::LocateDirectory);
+    if (dirNames.isEmpty()) {
         return QStringList();
     }
-    QDir customizationDir = QDir(customizationDirName);
-    const QStringList catalogFileFilters(QStringLiteral("catalog*.xml"));
-    const QFileInfoList catalogInfoFiles = customizationDir.entryInfoList(catalogFileFilters,
-                                           QDir::Files | QDir::NoSymLinks, QDir::Name);
     QStringList catalogFiles;
-    foreach (const QFileInfo &fileInfo, catalogInfoFiles) {
-        const QString fullFileName = QUrl::fromLocalFile(fileInfo.absoluteFilePath()).toEncoded();
-        if (fileInfo.fileName() == QStringLiteral("catalog.xml")) {
-           catalogFiles.prepend(fullFileName);
-        } else {
-           catalogFiles.append(fullFileName);
+    foreach (const QString &customizationDirName, dirNames) {
+        QDir customizationDir = QDir(customizationDirName);
+        const QStringList catalogFileFilters(QStringLiteral("catalog*.xml"));
+        const QFileInfoList catalogInfoFiles = customizationDir.entryInfoList(catalogFileFilters,
+                                               QDir::Files | QDir::NoSymLinks, QDir::Name);
+        foreach (const QFileInfo &fileInfo, catalogInfoFiles) {
+            const QString fullFileName = QUrl::fromLocalFile(fileInfo.absoluteFilePath()).toEncoded();
+            if (fileInfo.fileName() == QStringLiteral("catalog.xml")) {
+               catalogFiles.prepend(fullFileName);
+            } else {
+               catalogFiles.append(fullFileName);
+            }
         }
     }
 
diff --git a/src/xslt.h b/src/xslt.h
index 83d48b3..ceffadc 100644
--- a/src/xslt.h
+++ b/src/xslt.h
@@ -18,6 +18,7 @@ bool saveToCache(const QString &contents, const QString &filename);
 
 void setupStandardDirs(const QString &srcdir = QString());
 QString locateFileInDtdResource(const QString &file, const QStandardPaths::LocateOptions option=QStandardPaths::LocateFile);
+QStringList locateFilesInDtdResource(const QString &file, const QStandardPaths::LocateOptions option=QStandardPaths::LocateFile);
 QStringList getKDocToolsCatalogs();
 
 #endif
