diff --git a/CMakeLists.txt b/CMakeLists.txt
index 69e3798..e8fe059 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,6 +14,7 @@ find_package(QtGStreamer 1.0 REQUIRED)
 find_package(Qt4 COMPONENTS QtDeclarative)
 find_package(PkgConfig REQUIRED)
 find_package(KDeclarative REQUIRED)
+find_package(Phonon4Qt5 REQUIRED)
 
 include_directories(
     ${KDE4_INCLUDES}
diff --git a/libktpcall/CMakeLists.txt b/libktpcall/CMakeLists.txt
index 9d5a52b..1afb319 100644
--- a/libktpcall/CMakeLists.txt
+++ b/libktpcall/CMakeLists.txt
@@ -3,6 +3,7 @@ pkg_check_modules(FARSTREAM REQUIRED farstream-0.2)
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR}
         ${FARSTREAM_INCLUDE_DIRS}
+        ${PHONON_INCLUDE_DIR}
 )
 
 #####
@@ -31,6 +32,7 @@ target_link_libraries(ktpcall
     ${QTGSTREAMER_LIBRARIES}
     ${TELEPATHY_QT4_LIBRARIES}
     qtf
+    Phonon::phonon4qt5
 )
 
 #####
