diff -up kinit-5.10.0/src/klauncher/klauncher_main.cpp.qt_no_glib kinit-5.10.0/src/klauncher/klauncher_main.cpp
--- kinit-5.10.0/src/klauncher/klauncher_main.cpp.qt_no_glib	2015-05-03 10:14:51.000000000 -0500
+++ kinit-5.10.0/src/klauncher/klauncher_main.cpp	2015-05-20 08:49:34.594814571 -0500
@@ -152,10 +152,21 @@ extern "C" Q_DECL_EXPORT int kdemain(int
     // WABA: Make sure not to enable session management.
     putenv(strdup("SESSION_MANAGER="));
 
+    // Disable the GLib event loop (rh#983110)
+    bool wasQtNoGlibSet = (getenv("QT_NO_GLIB") != 0);
+    if (!wasQtNoGlibSet) {
+       setenv("QT_NO_GLIB", "1", true);
+    }
+
     // We need a QGuiApplication as we use X11
     QGuiApplication app(argc, argv);
     app.setApplicationName(QStringLiteral("klauncher"));
 
+    // Now get rid of QT_NO_GLIB again so launched processes don't inherit it
+    if (!wasQtNoGlibSet) {
+       unsetenv("QT_NO_GLIB");
+    }
+
     int maxTry = 3;
     while (true) {
         QString service(QLatin1String("org.kde.klauncher5")); // same as ktoolinvocation.cpp
