diff --git a/archive/CMakeLists.txt b/archive/CMakeLists.txt
index ec2cf7ad6c9c645e62a47c94ec617f8a37c5f708..83e02b36ed27a6fbdc82ffb3eba1d21cbee737d9 100644
--- a/archive/CMakeLists.txt
+++ b/archive/CMakeLists.txt
@@ -2,8 +2,9 @@ add_subdirectory(tests)
 
 add_library(kio_archive MODULE kio_archive.cpp)
 target_link_libraries(kio_archive KF5::KIOCore KF5::Archive KF5::KDE4Support)
+set_target_properties(kio_archive PROPERTIES OUTPUT_NAME "archive")
 
-install(TARGETS kio_archive DESTINATION ${PLUGIN_INSTALL_DIR} )
-install( FILES tar.protocol ar.protocol zip.protocol  DESTINATION  ${SERVICES_INSTALL_DIR} )
+install(TARGETS kio_archive DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio )
+install(FILES tar.protocol ar.protocol zip.protocol DESTINATION ${SERVICES_INSTALL_DIR})
 
 
diff --git a/archive/ar.protocol b/archive/ar.protocol
index 7a848e505e5f00121e8ec5493564dfacdd405c31..70db371d79d13d3f73c72afbd49f84ab88782d0c 100644
--- a/archive/ar.protocol
+++ b/archive/ar.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_archive
+exec=kf5/kio/archive
 protocol=ar
 archiveMimetype=application/x-archive
 input=filesystem
diff --git a/archive/tar.protocol b/archive/tar.protocol
index 19447a5e75b106e7a8c8e8396375bb1fc600a6da..c713eac548c3c38ef82a929ad3710f8746dea94d 100644
--- a/archive/tar.protocol
+++ b/archive/tar.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_archive
+exec=kf5/kio/archive
 protocol=tar
 archiveMimetype=application/x-tar,application/x-compressed-tar,application/x-bzip-compressed-tar,application/x-webarchive,application/x-lzma-compressed-tar,application/x-xz-compressed-tar
 input=filesystem
diff --git a/archive/zip.protocol b/archive/zip.protocol
index ce7c54b4e92dfab29150246bf6588d1f8bb49057..78310e6c95f13ea1e25d89e31852decb13dfe33b 100644
--- a/archive/zip.protocol
+++ b/archive/zip.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_archive
+exec=kf5/kio/archive
 protocol=zip
 archiveMimetype=application/zip
 input=filesystem
diff --git a/bookmarks/CMakeLists.txt b/bookmarks/CMakeLists.txt
index 703b109c2e049edecce1742482bd82e5c92caecd..9eca94b59b0998373ba134af7fa86a383c4d639e 100644
--- a/bookmarks/CMakeLists.txt
+++ b/bookmarks/CMakeLists.txt
@@ -6,7 +6,9 @@ add_library(kio_bookmarks MODULE ${kio_bookmarks_PART_SRCS})
 
 target_link_libraries(kio_bookmarks KF5::KIOCore KF5::KDE4Support KF5::Solid)
 
-install(TARGETS kio_bookmarks  DESTINATION ${PLUGIN_INSTALL_DIR} )
+set_target_properties(kio_bookmarks PROPERTIES OUTPUT_NAME "bookmarks")
+
+install(TARGETS kio_bookmarks DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 
 
 ########### install files ###############
diff --git a/bookmarks/bookmarks.protocol b/bookmarks/bookmarks.protocol
index 0642bd2f7277be8707351582f219281d45834fa6..0898a6b38340b1b76a6785dcf786fb7e4b45b0ab 100644
--- a/bookmarks/bookmarks.protocol
+++ b/bookmarks/bookmarks.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_bookmarks
+exec=kf5/kio/bookmarks
 protocol=bookmarks
 input=none
 output=stream
diff --git a/desktop/CMakeLists.txt b/desktop/CMakeLists.txt
index f7bd6709340a5f4e7e5f6fbd076400a76b526cd8..ac9968731e87ff337c7798090ffc3d06bc3ada0d 100644
--- a/desktop/CMakeLists.txt
+++ b/desktop/CMakeLists.txt
@@ -4,7 +4,9 @@ add_library(kio_desktop MODULE kio_desktop.cpp)
 
 target_link_libraries(kio_desktop KF5::KIOCore KF5::KDE4Support)
 
-install(TARGETS kio_desktop DESTINATION  ${PLUGIN_INSTALL_DIR} )
+set_target_properties(kio_desktop PROPERTIES OUTPUT_NAME "desktop")
+
+install(TARGETS kio_desktop DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 
 ########### next target ###############
 
diff --git a/desktop/desktop.protocol b/desktop/desktop.protocol
index 0e388386f1794dd90d445ec146ef54a04f595c8b..c6abe5eb91d7bb6a1b5b65fca03a56f774c5fabd 100644
--- a/desktop/desktop.protocol
+++ b/desktop/desktop.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_desktop
+exec=kf5/kio/desktop
 protocol=desktop
 Icon=user-desktop
 input=none
diff --git a/filter/CMakeLists.txt b/filter/CMakeLists.txt
index 240cc7b57bd38e86fd50084e53c17f1672b2b946..0d5d2e2223cbeb889773d2f393c3b2cafa3646bd 100644
--- a/filter/CMakeLists.txt
+++ b/filter/CMakeLists.txt
@@ -16,8 +16,10 @@ set_package_properties(LibLZMA PROPERTIES DESCRIPTION "A very high compression r
 
 add_library(kio_filter MODULE filter.cc)
 target_link_libraries(kio_filter KF5::KDE4Support KF5::Archive KF5::KIOCore)
+set_target_properties(kio_filter PROPERTIES OUTPUT_NAME "filter")
 
-install(TARGETS kio_filter DESTINATION ${PLUGIN_INSTALL_DIR})
+
+install(TARGETS kio_filter DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 install( FILES gzip.protocol  DESTINATION  ${SERVICES_INSTALL_DIR} )
 
 if(BZIP2_FOUND)
diff --git a/filter/bzip.protocol b/filter/bzip.protocol
index 7d3cb570fbc534fbba22dd8e10472b47053c7195..97c8669a895e273f71e46363c1d4b57f2f25be5c 100644
--- a/filter/bzip.protocol
+++ b/filter/bzip.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_filter
+exec=kf5/kio/filter
 protocol=bzip
 archiveMimetype=application/x-bzip
 determineMimetypeFromExtension=false
diff --git a/filter/bzip2.protocol b/filter/bzip2.protocol
index 1baaf7dd610f39325b2aa21bdee5bf1e63e5835f..ab274024e959060e61e7e375d1d285ff12dd6040 100644
--- a/filter/bzip2.protocol
+++ b/filter/bzip2.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_filter
+exec=kf5/kio/filter
 protocol=bzip2
 archiveMimetype=application/x-bzip2
 determineMimetypeFromExtension=false
diff --git a/filter/gzip.protocol b/filter/gzip.protocol
index 8ed55ec16f49f42fe6d419327484b33d59ec537b..9a0d2894ced4e8919ac43e67202422929544f56a 100644
--- a/filter/gzip.protocol
+++ b/filter/gzip.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_filter
+exec=kf5/kio/filter
 protocol=gzip
 archiveMimetype=application/gzip
 determineMimetypeFromExtension=false
diff --git a/filter/lzma.protocol b/filter/lzma.protocol
index 13bb28e3d870f8130d052efd97a5a688d9133c31..b6c59aeb75ccad5342c3fcb8a6475a19cf65ea03 100644
--- a/filter/lzma.protocol
+++ b/filter/lzma.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_filter
+exec=kf5/kio/filter
 protocol=lzma
 archiveMimetype=application/x-lzma
 determineMimetypeFromExtension=true
diff --git a/filter/xz.protocol b/filter/xz.protocol
index f4fb7f01146cbf4d5afce528462e0c5dafa9591e..9a631c248d5c129f4172cfa945c176cf6344ee4b 100644
--- a/filter/xz.protocol
+++ b/filter/xz.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_filter
+exec=kf5/kio/filter
 protocol=xz
 archiveMimetype=application/x-xz
 determineMimetypeFromExtension=false
diff --git a/fish/CMakeLists.txt b/fish/CMakeLists.txt
index 24136d4dac01a8a08db8d17380661a9051fa0b04..1a73b9d9ec39e697f925fd823efa456c476af7db 100644
--- a/fish/CMakeLists.txt
+++ b/fish/CMakeLists.txt
@@ -45,11 +45,13 @@ if (MD5SUM_EXECUTABLE)
 
    target_link_libraries(kio_fish KF5::KIOCore KF5::KDE4Support)
 
+   set_target_properties(kio_fish PROPERTIES OUTPUT_NAME "fish")
+
    if (UTIL_LIBRARIES)
       target_link_libraries(kio_fish ${UTIL_LIBRARIES})
    endif ()
 
-   install(TARGETS kio_fish  DESTINATION ${PLUGIN_INSTALL_DIR} )
+   install(TARGETS kio_fish  DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio )
 
 
    ########### install files ###############
diff --git a/fish/fish.protocol b/fish/fish.protocol
index ff5784eea877d0fd9a3c6ac4724056ddfd8fe5f2..7a053ddd6c79d7b7a37cc662663bc1561c8e0540 100644
--- a/fish/fish.protocol
+++ b/fish/fish.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_fish
+exec=kf5/kio/fish
 protocol=fish
 input=none
 output=filesystem
diff --git a/info/CMakeLists.txt b/info/CMakeLists.txt
index 0163e82c4f865e3fb3b09bf5ddd44e272ce9c609..7d71b06a8bc0b77a857ab429c37cb757542580be 100644
--- a/info/CMakeLists.txt
+++ b/info/CMakeLists.txt
@@ -4,8 +4,9 @@ target_link_libraries(kio_info
     KF5::IconThemes
     KF5::KIOCore
 )
+set_target_properties(kio_info PROPERTIES OUTPUT_NAME "info")
 
-install(TARGETS kio_info  DESTINATION ${PLUGIN_INSTALL_DIR} )
+install(TARGETS kio_info  DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio )
 install( FILES kde-info2html.conf DESTINATION  ${DATA_INSTALL_DIR}/kio_info )
 install( FILES info.protocol  DESTINATION  ${SERVICES_INSTALL_DIR} )
 install( PROGRAMS kde-info2html DESTINATION  ${DATA_INSTALL_DIR}/kio_info )
diff --git a/info/info.protocol b/info/info.protocol
index fa8cbb4f6eaaeeaf8bfdead0fad598df8fe588eb..a8d563a64e39ff42f781d3ff3aa7dd63c9df2eab 100644
--- a/info/info.protocol
+++ b/info/info.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_info
+exec=kf5/kio/info
 protocol=info
 input=none
 output=filesystem
diff --git a/man/CMakeLists.txt b/man/CMakeLists.txt
index 3343ed85978dbbfa72a5f2cc8530d0f0c043df48..cf6bda60668c4a5398a0920c4f0c6de5302932f0 100644
--- a/man/CMakeLists.txt
+++ b/man/CMakeLists.txt
@@ -6,7 +6,9 @@ set(kio_man_PART_SRCS man2html.cpp kio_man.cpp request_hash.cpp )
 add_library(kio_man MODULE ${kio_man_PART_SRCS})
 
 target_link_libraries(kio_man KF5::KIOCore Qt5::Widgets KF5::Archive KF5::KDE4Support)
-install(TARGETS kio_man  DESTINATION ${PLUGIN_INSTALL_DIR} )
+set_target_properties(kio_man PROPERTIES OUTPUT_NAME "man")
+
+install(TARGETS kio_man  DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio )
 
 ########### next target ###############
 
diff --git a/man/man.protocol b/man/man.protocol
index cc100ce5ec48b40c890abe5d7afdf02ef589810b..42799e471035f19265faa397e0a7027de035bb84 100644
--- a/man/man.protocol
+++ b/man/man.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_man
+exec=kf5/kio/man
 protocol=man
 input=none
 output=filesystem
diff --git a/network/ioslave/CMakeLists.txt b/network/ioslave/CMakeLists.txt
index 67fc482cd390f35fb2d5c1740994d2fe9db254f1..06a964dda284cf84c8a581b3b60663d6b569b192 100644
--- a/network/ioslave/CMakeLists.txt
+++ b/network/ioslave/CMakeLists.txt
@@ -17,10 +17,10 @@ set( kio_network_SRCS
 # qt5_add_dbus_interface( kio_network_SRCS  ../kded/org.kde.network.xml networkdbusproxy )
 
 add_library(kio_network MODULE  ${kio_network_SRCS} )
-
 target_link_libraries( kio_network  molletnetwork KF5::KIOCore )
+set_target_properties(kio_network PROPERTIES OUTPUT_NAME "network")
 
-install( TARGETS kio_network  DESTINATION ${PLUGIN_INSTALL_DIR})
+install( TARGETS kio_network  DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 
 install( FILES network.protocol  DESTINATION ${SERVICES_INSTALL_DIR})
 install( FILES network.desktop  DESTINATION ${DATA_INSTALL_DIR}/remoteview)
diff --git a/network/ioslave/network.protocol b/network/ioslave/network.protocol
index 1e10cc6c3a6aadf2ee6794ab36200ef6793e324e..b648e2e40f3fbc27ceded07acbc5469d5264bf5e 100644
--- a/network/ioslave/network.protocol
+++ b/network/ioslave/network.protocol
@@ -78,7 +78,7 @@ Description[wa]=On vårlet d' I/R po betchter el rantoele
 Description[x-test]=xxA kioslave to browse the networkxx
 Description[zh_CN]=浏览网络的 KIO slave
 Description[zh_TW]=瀏覽網路的 kioslave
-exec=kio_network
+exec=kf5/kio/network
 
 reading=true
 listing=Name
diff --git a/nfs/CMakeLists.txt b/nfs/CMakeLists.txt
index dfc6eae3fad51eee0e736970da9520959eeed1f5..93f59b8d01b58c46420e56d9f80b2d7af0291cd4 100644
--- a/nfs/CMakeLists.txt
+++ b/nfs/CMakeLists.txt
@@ -1,7 +1,8 @@
 add_library(kio_nfs MODULE kio_nfs.cpp mount_xdr.c nfs_prot_xdr.c)
 target_link_libraries(kio_nfs KF5::KIOCore KF5::I18n)
+set_target_properties(kio_nfs PROPERTIES OUTPUT_NAME "nfs")
 
-install(TARGETS kio_nfs  DESTINATION ${PLUGIN_INSTALL_DIR} )
+install(TARGETS kio_nfs DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 
 ########### install files ###############
 
diff --git a/nfs/nfs.protocol b/nfs/nfs.protocol
index 85cf2035e44bf7bfe65d4e2b62d1f44fa05d6001..3c59facdf0934be6242d87faba5137d523449c84 100644
--- a/nfs/nfs.protocol
+++ b/nfs/nfs.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_nfs
+exec=kf5/kio/nfs
 protocol=nfs
 input=none
 output=filesystem
diff --git a/recentdocuments/CMakeLists.txt b/recentdocuments/CMakeLists.txt
index bc2b9dbf35428cf0dd1e6ad5c6e569806c92d47f..719e7650a36a8ea38ab0e1031bf6a82be8da478a 100644
--- a/recentdocuments/CMakeLists.txt
+++ b/recentdocuments/CMakeLists.txt
@@ -1,7 +1,8 @@
 add_library(kio_recentdocuments MODULE recentdocuments.cpp)
 target_link_libraries(kio_recentdocuments KF5::KIOCore Qt5::DBus KF5::KDE4Support)
+set_target_properties(kio_recentdocuments PROPERTIES OUTPUT_NAME "recentdocuments")
 
-install(TARGETS kio_recentdocuments  DESTINATION ${PLUGIN_INSTALL_DIR})
+install(TARGETS kio_recentdocuments  DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 
 ########### next target ###############
 
diff --git a/recentdocuments/recentdocuments.protocol b/recentdocuments/recentdocuments.protocol
index c0bf1ebd80a2c696cad3062ef047ac4847270fc5..dc6e86f9939b2887be3eb5a91e8aafa22a481a3e 100644
--- a/recentdocuments/recentdocuments.protocol
+++ b/recentdocuments/recentdocuments.protocol
@@ -1,6 +1,6 @@
 [Protocol]
 DocPath=kioslave/kio_recentdocuments.html
-exec=kio_recentdocuments
+exec=kf5/kio/recentdocuments
 protocol=recentdocuments
 Icon=document-open-recent
 input=none
diff --git a/settings/CMakeLists.txt b/settings/CMakeLists.txt
index 5a57a1840fb33b32701a6ecfff24c42739b05c45..8dc0c64181657a75cc11a61eb5a43064ed71e895 100644
--- a/settings/CMakeLists.txt
+++ b/settings/CMakeLists.txt
@@ -1,6 +1,7 @@
 add_library(kio_settings MODULE kio_settings.cpp)
-
 target_link_libraries(kio_settings KF5::KDE4Support KF5::KIOCore)
+set_target_properties(kio_settings PROPERTIES OUTPUT_NAME "settings")
+
 
-install(TARGETS kio_settings  DESTINATION ${PLUGIN_INSTALL_DIR} )
+install(TARGETS kio_settings  DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio )
 install( FILES settings.protocol DESTINATION  ${SERVICES_INSTALL_DIR} )
diff --git a/settings/settings.protocol b/settings/settings.protocol
index efb03e865594f30be5ed0d1013d31a0200eed816..002512ac32865a4e2a8c5e6e331f378f4a8c4ac5 100644
--- a/settings/settings.protocol
+++ b/settings/settings.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_settings
+exec=kf5/kio/settings
 protocol=settings
 input=none
 output=filesystem
diff --git a/sftp/CMakeLists.txt b/sftp/CMakeLists.txt
index defb7dd041ebfd5af0ac19a252cc9f2f454f67f9..44b20930ead880beaca3c1ff895d8cf84afc20bb 100644
--- a/sftp/CMakeLists.txt
+++ b/sftp/CMakeLists.txt
@@ -2,8 +2,9 @@ include_directories(${LIBSSH_INCLUDE_DIR})
 
 add_library(kio_sftp MODULE kio_sftp.cpp)
 target_link_libraries(kio_sftp KF5::KIOCore ${LIBSSH_LIBRARIES} KF5::KDE4Support)
+set_target_properties(kio_sftp PROPERTIES OUTPUT_NAME "sftp")
 
-install(TARGETS kio_sftp  DESTINATION ${PLUGIN_INSTALL_DIR} )
+install(TARGETS kio_sftp  DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio )
 
 install( FILES sftp.protocol  DESTINATION  ${SERVICES_INSTALL_DIR} )
 
diff --git a/sftp/sftp.protocol b/sftp/sftp.protocol
index ec15eeb26b3ce2efded173f4c7a4cd737e0a05f8..6a45c08f9ea6359f525ee5661e2d92746c6d4be6 100644
--- a/sftp/sftp.protocol
+++ b/sftp/sftp.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_sftp
+exec=kf5/kio/sftp
 protocol=sftp
 input=none
 listing=Name,Type,Size,Date,AccessDate,Access,Owner,Group,Link,MimeType
diff --git a/smb/CMakeLists.txt b/smb/CMakeLists.txt
index a3a772f3028995b811b01fcebe11d8715d3e1620..406a22786cfbb6826895fa1281256fb33ff3876d 100644
--- a/smb/CMakeLists.txt
+++ b/smb/CMakeLists.txt
@@ -31,7 +31,9 @@ add_library(kio_smb MODULE ${kio_smb_PART_SRCS})
 target_link_libraries(kio_smb KF5::KIOCore KF5::I18n mpr)    
 endif()
 
-install(TARGETS kio_smb  DESTINATION ${PLUGIN_INSTALL_DIR} )
+set_target_properties(kio_smb PROPERTIES OUTPUT_NAME "smb")
+
+install(TARGETS kio_smb DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 
 
 ########### install files ###############
diff --git a/smb/smb.protocol b/smb/smb.protocol
index e58597844ff8554e837f6ee8807d17820e377945..8a019649bc69ef784b1e30012cf1bc615250ec48 100644
--- a/smb/smb.protocol
+++ b/smb/smb.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_smb
+exec=kf5/kio/smb
 protocol=smb
 input=none
 output=filesystem
diff --git a/thumbnail/CMakeLists.txt b/thumbnail/CMakeLists.txt
index f3733d452eb133d2b5a1445740db0312cd5d8251..9363fab559b0968c3eeb0f308a8f6017c5e259bf 100644
--- a/thumbnail/CMakeLists.txt
+++ b/thumbnail/CMakeLists.txt
@@ -43,7 +43,9 @@ target_link_libraries(kio_thumbnail
 	KF5::KDE4Support
 )
 
-install(TARGETS kio_thumbnail DESTINATION ${PLUGIN_INSTALL_DIR})
+set_target_properties(kio_thumbnail PROPERTIES OUTPUT_NAME "thumbnail")
+
+install(TARGETS kio_thumbnail DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 
 ########### next target ###############
 
diff --git a/thumbnail/thumbnail.protocol b/thumbnail/thumbnail.protocol
index eef743ab5310007093e6a1c1ff6d8c258f1be5d9..360e68fecd8e61d44e5d393b554924907f0076e8 100644
--- a/thumbnail/thumbnail.protocol
+++ b/thumbnail/thumbnail.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_thumbnail
+exec=kf5/kio/thumbnail
 protocol=thumbnail
 input=stream
 output=stream
diff --git a/trash/CMakeLists.txt b/trash/CMakeLists.txt
index 4ee03581f5d632282efb9b8902b3abb0e95d169c..e9e2964759e3ad6bfa04576d710d9aba3148f0d0 100644
--- a/trash/CMakeLists.txt
+++ b/trash/CMakeLists.txt
@@ -29,13 +29,16 @@ endif()
 add_library(kio_trash MODULE ${kio_trash_PART_SRCS})
 
 target_link_libraries(kio_trash
-	KF5::Solid
+        KF5::Solid
         KF5::KDE4Support
-	KF5::KIOCore
+        KF5::KIOCore
         Qt5::DBus
-        KF5::I18n KF5::ConfigCore KF5::ConfigGui 
-        )
-install(TARGETS kio_trash  DESTINATION ${PLUGIN_INSTALL_DIR} )
+        KF5::I18n KF5::ConfigCore KF5::ConfigGui
+)
+
+set_target_properties(kio_trash PROPERTIES OUTPUT_NAME "trash")
+
+install(TARGETS kio_trash  DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio )
 
 set(ktrash_SRCS ktrash.cpp )
 
@@ -44,9 +47,9 @@ ecm_mark_nongui_executable(ktrash5)
 
 target_link_libraries(ktrash5
         KF5::KDE4Support
-	KF5::KIOCore
+        KF5::KIOCore
         KF5::I18n KF5::ConfigCore KF5::ConfigGui
-        )
+)
 
 install(TARGETS ktrash5 ${INSTALL_TARGETS_DEFAULT_ARGS})
 
diff --git a/trash/trash.protocol b/trash/trash.protocol
index 277698579c8a70bac5cadff5cc1f379053e4c61e..b8b87b983c72f62dd4c54ec40737e2e4b8f3276f 100644
--- a/trash/trash.protocol
+++ b/trash/trash.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_trash
+exec=kf5/kio/trash
 protocol=trash
 input=none
 output=filesystem
