diff --git a/archive/CMakeLists.txt b/archive/CMakeLists.txt
index 0524aec..6170b75 100644
--- a/archive/CMakeLists.txt
+++ b/archive/CMakeLists.txt
@@ -4,6 +4,7 @@ add_subdirectory(tests)
 
 add_library(kio_archive MODULE kio_archive.cpp)
 target_link_libraries(kio_archive KF5::KIOCore KF5::Archive KF5::KDE4Support)
+set_target_properties(kio_archive PROPERTIES OUTPUT_NAME "archive")
 
-install(TARGETS kio_archive DESTINATION ${PLUGIN_INSTALL_DIR} )
+install(TARGETS kio_archive DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio )
 install( FILES tar.protocol ar.protocol zip.protocol  DESTINATION  ${SERVICES_INSTALL_DIR} )
diff --git a/archive/ar.protocol b/archive/ar.protocol
index 7a848e5..70db371 100644
--- a/archive/ar.protocol
+++ b/archive/ar.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_archive
+exec=kf5/kio/archive
 protocol=ar
 archiveMimetype=application/x-archive
 input=filesystem
diff --git a/archive/tar.protocol b/archive/tar.protocol
index 19447a5..c713eac 100644
--- a/archive/tar.protocol
+++ b/archive/tar.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_archive
+exec=kf5/kio/archive
 protocol=tar
 archiveMimetype=application/x-tar,application/x-compressed-tar,application/x-bzip-compressed-tar,application/x-webarchive,application/x-lzma-compressed-tar,application/x-xz-compressed-tar
 input=filesystem
diff --git a/archive/zip.protocol b/archive/zip.protocol
index ce7c54b..78310e6 100644
--- a/archive/zip.protocol
+++ b/archive/zip.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_archive
+exec=kf5/kio/archive
 protocol=zip
 archiveMimetype=application/zip
 input=filesystem
diff --git a/bookmarks/CMakeLists.txt b/bookmarks/CMakeLists.txt
index c44355c..d09da32 100644
--- a/bookmarks/CMakeLists.txt
+++ b/bookmarks/CMakeLists.txt
@@ -7,7 +7,9 @@ add_library(kio_bookmarks MODULE ${kio_bookmarks_PART_SRCS})
 
 target_link_libraries(kio_bookmarks KF5::KIOCore KF5::KDE4Support KF5::Solid)
 
-install(TARGETS kio_bookmarks  DESTINATION ${PLUGIN_INSTALL_DIR} )
+set_target_properties(kio_bookmarks PROPERTIES OUTPUT_NAME "bookmarks")
+
+install(TARGETS kio_bookmarks DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 
 
 ########### install files ###############
diff --git a/bookmarks/bookmarks.protocol b/bookmarks/bookmarks.protocol
index 0642bd2..0898a6b 100644
--- a/bookmarks/bookmarks.protocol
+++ b/bookmarks/bookmarks.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_bookmarks
+exec=kf5/kio/bookmarks
 protocol=bookmarks
 input=none
 output=stream
diff --git a/desktop/CMakeLists.txt b/desktop/CMakeLists.txt
index f7bd670..ac99687 100644
--- a/desktop/CMakeLists.txt
+++ b/desktop/CMakeLists.txt
@@ -4,7 +4,9 @@ add_library(kio_desktop MODULE kio_desktop.cpp)
 
 target_link_libraries(kio_desktop KF5::KIOCore KF5::KDE4Support)
 
-install(TARGETS kio_desktop DESTINATION  ${PLUGIN_INSTALL_DIR} )
+set_target_properties(kio_desktop PROPERTIES OUTPUT_NAME "desktop")
+
+install(TARGETS kio_desktop DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 
 ########### next target ###############
 
diff --git a/desktop/desktop.protocol b/desktop/desktop.protocol
index 0e38838..c6abe5e 100644
--- a/desktop/desktop.protocol
+++ b/desktop/desktop.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_desktop
+exec=kf5/kio/desktop
 protocol=desktop
 Icon=user-desktop
 input=none
diff --git a/filter/CMakeLists.txt b/filter/CMakeLists.txt
index f532d0f..3e8a200 100644
--- a/filter/CMakeLists.txt
+++ b/filter/CMakeLists.txt
@@ -1,7 +1,8 @@
 add_library(kio_filter MODULE filter.cc)
 target_link_libraries(kio_filter KF5::KDE4Support KF5::Archive KF5::KIOCore)
+set_target_properties(kio_filter PROPERTIES OUTPUT_NAME "filter")
 
-install(TARGETS kio_filter DESTINATION ${PLUGIN_INSTALL_DIR})
+install(TARGETS kio_filter DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 install( FILES gzip.protocol  DESTINATION  ${SERVICES_INSTALL_DIR} )
 
 if(KArchive_HAVE_BZIP2)
diff --git a/filter/bzip.protocol b/filter/bzip.protocol
index 7d3cb57..97c8669 100644
--- a/filter/bzip.protocol
+++ b/filter/bzip.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_filter
+exec=kf5/kio/filter
 protocol=bzip
 archiveMimetype=application/x-bzip
 determineMimetypeFromExtension=false
diff --git a/filter/bzip2.protocol b/filter/bzip2.protocol
index 1baaf7d..ab27402 100644
--- a/filter/bzip2.protocol
+++ b/filter/bzip2.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_filter
+exec=kf5/kio/filter
 protocol=bzip2
 archiveMimetype=application/x-bzip2
 determineMimetypeFromExtension=false
diff --git a/filter/gzip.protocol b/filter/gzip.protocol
index 8ed55ec..9a0d289 100644
--- a/filter/gzip.protocol
+++ b/filter/gzip.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_filter
+exec=kf5/kio/filter
 protocol=gzip
 archiveMimetype=application/gzip
 determineMimetypeFromExtension=false
diff --git a/filter/lzma.protocol b/filter/lzma.protocol
index 13bb28e..b6c59ae 100644
--- a/filter/lzma.protocol
+++ b/filter/lzma.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_filter
+exec=kf5/kio/filter
 protocol=lzma
 archiveMimetype=application/x-lzma
 determineMimetypeFromExtension=true
diff --git a/filter/xz.protocol b/filter/xz.protocol
index f4fb7f0..9a631c2 100644
--- a/filter/xz.protocol
+++ b/filter/xz.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_filter
+exec=kf5/kio/filter
 protocol=xz
 archiveMimetype=application/x-xz
 determineMimetypeFromExtension=false
diff --git a/fish/CMakeLists.txt b/fish/CMakeLists.txt
index 6b0c168..ca75771 100644
--- a/fish/CMakeLists.txt
+++ b/fish/CMakeLists.txt
@@ -47,11 +47,13 @@ if (MD5SUM_EXECUTABLE)
 
    target_link_libraries(kio_fish KF5::KIOCore KF5::KDE4Support)
 
+   set_target_properties(kio_fish PROPERTIES OUTPUT_NAME "fish")
+
    if (UTIL_LIBRARIES)
       target_link_libraries(kio_fish ${UTIL_LIBRARIES})
    endif ()
 
-   install(TARGETS kio_fish  DESTINATION ${PLUGIN_INSTALL_DIR} )
+   install(TARGETS kio_fish  DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio )
 
 
    ########### install files ###############
diff --git a/fish/fish.protocol b/fish/fish.protocol
index ff5784e..7a053dd 100644
--- a/fish/fish.protocol
+++ b/fish/fish.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_fish
+exec=kf5/kio/fish
 protocol=fish
 input=none
 output=filesystem
diff --git a/info/CMakeLists.txt b/info/CMakeLists.txt
index d9541b5..40e10ce 100644
--- a/info/CMakeLists.txt
+++ b/info/CMakeLists.txt
@@ -6,8 +6,9 @@ target_link_libraries(kio_info
     KF5::IconThemes
     KF5::KIOCore
 )
+set_target_properties(kio_info PROPERTIES OUTPUT_NAME "info")
 
-install(TARGETS kio_info  DESTINATION ${PLUGIN_INSTALL_DIR} )
+install(TARGETS kio_info  DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio )
 install( FILES kde-info2html.conf DESTINATION  ${DATA_INSTALL_DIR}/kio_info )
 install( FILES info.protocol  DESTINATION  ${SERVICES_INSTALL_DIR} )
 install( PROGRAMS kde-info2html DESTINATION  ${DATA_INSTALL_DIR}/kio_info )
diff --git a/info/info.protocol b/info/info.protocol
index fa8cbb4..a8d563a 100644
--- a/info/info.protocol
+++ b/info/info.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_info
+exec=kf5/kio/info
 protocol=info
 input=none
 output=filesystem
diff --git a/man/CMakeLists.txt b/man/CMakeLists.txt
index 1aa7b24..77d2028 100644
--- a/man/CMakeLists.txt
+++ b/man/CMakeLists.txt
@@ -8,7 +8,9 @@ set(kio_man_PART_SRCS man2html.cpp kio_man.cpp request_hash.cpp )
 add_library(kio_man MODULE ${kio_man_PART_SRCS})
 
 target_link_libraries(kio_man KF5::KIOCore Qt5::Widgets KF5::Archive KF5::KDE4Support)
-install(TARGETS kio_man  DESTINATION ${PLUGIN_INSTALL_DIR} )
+set_target_properties(kio_man PROPERTIES OUTPUT_NAME "man")
+
+install(TARGETS kio_man  DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio )
 
 ########### next target ###############
 
diff --git a/man/man.protocol b/man/man.protocol
index cc100ce..42799e4 100644
--- a/man/man.protocol
+++ b/man/man.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_man
+exec=kf5/kio/man
 protocol=man
 input=none
 output=filesystem
diff --git a/network/ioslave/CMakeLists.txt b/network/ioslave/CMakeLists.txt
index 67fc482..06a964d 100644
--- a/network/ioslave/CMakeLists.txt
+++ b/network/ioslave/CMakeLists.txt
@@ -17,10 +17,10 @@ set( kio_network_SRCS
 # qt5_add_dbus_interface( kio_network_SRCS  ../kded/org.kde.network.xml networkdbusproxy )
 
 add_library(kio_network MODULE  ${kio_network_SRCS} )
-
 target_link_libraries( kio_network  molletnetwork KF5::KIOCore )
+set_target_properties(kio_network PROPERTIES OUTPUT_NAME "network")
 
-install( TARGETS kio_network  DESTINATION ${PLUGIN_INSTALL_DIR})
+install( TARGETS kio_network  DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 
 install( FILES network.protocol  DESTINATION ${SERVICES_INSTALL_DIR})
 install( FILES network.desktop  DESTINATION ${DATA_INSTALL_DIR}/remoteview)
diff --git a/network/ioslave/network.protocol b/network/ioslave/network.protocol
index 1e10cc6..b648e2e 100644
--- a/network/ioslave/network.protocol
+++ b/network/ioslave/network.protocol
@@ -78,7 +78,7 @@ Description[wa]=On vårlet d' I/R po betchter el rantoele
 Description[x-test]=xxA kioslave to browse the networkxx
 Description[zh_CN]=浏览网络的 KIO slave
 Description[zh_TW]=瀏覽網路的 kioslave
-exec=kio_network
+exec=kf5/kio/network
 
 reading=true
 listing=Name
diff --git a/nfs/CMakeLists.txt b/nfs/CMakeLists.txt
index 7987427..2844d74 100644
--- a/nfs/CMakeLists.txt
+++ b/nfs/CMakeLists.txt
@@ -2,8 +2,9 @@ add_definitions(-DTRANSLATION_DOMAIN=\"kio_nfs\")
 
 add_library(kio_nfs MODULE kio_nfs.cpp mount_xdr.c nfs_prot_xdr.c)
 target_link_libraries(kio_nfs KF5::KIOCore KF5::I18n Qt5::Network)
+set_target_properties(kio_nfs PROPERTIES OUTPUT_NAME "nfs")
 
-install(TARGETS kio_nfs  DESTINATION ${PLUGIN_INSTALL_DIR} )
+install(TARGETS kio_nfs  DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio )
 
 ########### install files ###############
 
diff --git a/nfs/nfs.protocol b/nfs/nfs.protocol
index 85cf203..3c59fac 100644
--- a/nfs/nfs.protocol
+++ b/nfs/nfs.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_nfs
+exec=kf5/kio/nfs
 protocol=nfs
 input=none
 output=filesystem
diff --git a/recentdocuments/CMakeLists.txt b/recentdocuments/CMakeLists.txt
index 85d0c5c..539ee0f 100644
--- a/recentdocuments/CMakeLists.txt
+++ b/recentdocuments/CMakeLists.txt
@@ -2,8 +2,9 @@ add_definitions(-DTRANSLATION_DOMAIN=\"kio_recentdocuments\")
 
 add_library(kio_recentdocuments MODULE recentdocuments.cpp)
 target_link_libraries(kio_recentdocuments KF5::KIOCore Qt5::DBus KF5::KDE4Support)
+set_target_properties(kio_recentdocuments PROPERTIES OUTPUT_NAME "recentdocuments")
 
-install(TARGETS kio_recentdocuments  DESTINATION ${PLUGIN_INSTALL_DIR})
+install(TARGETS kio_recentdocuments  DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 
 ########### next target ###############
 
diff --git a/recentdocuments/recentdocuments.protocol b/recentdocuments/recentdocuments.protocol
index c0bf1eb..dc6e86f 100644
--- a/recentdocuments/recentdocuments.protocol
+++ b/recentdocuments/recentdocuments.protocol
@@ -1,6 +1,6 @@
 [Protocol]
 DocPath=kioslave/kio_recentdocuments.html
-exec=kio_recentdocuments
+exec=kf5/kio/recentdocuments
 protocol=recentdocuments
 Icon=document-open-recent
 input=none
diff --git a/settings/CMakeLists.txt b/settings/CMakeLists.txt
index a318e9b..fe56fa6 100644
--- a/settings/CMakeLists.txt
+++ b/settings/CMakeLists.txt
@@ -1,8 +1,9 @@
 add_definitions(-DTRANSLATION_DOMAIN=\"kio_settings\")
 
 add_library(kio_settings MODULE kio_settings.cpp)
-
 target_link_libraries(kio_settings KF5::KDE4Support KF5::KIOCore)
+set_target_properties(kio_settings PROPERTIES OUTPUT_NAME "settings")
+
 
-install(TARGETS kio_settings  DESTINATION ${PLUGIN_INSTALL_DIR} )
+install(TARGETS kio_settings  DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio )
 install( FILES settings.protocol DESTINATION  ${SERVICES_INSTALL_DIR} )
diff --git a/settings/settings.protocol b/settings/settings.protocol
index efb03e8..002512a 100644
--- a/settings/settings.protocol
+++ b/settings/settings.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_settings
+exec=kf5/kio/settings
 protocol=settings
 input=none
 output=filesystem
diff --git a/sftp/CMakeLists.txt b/sftp/CMakeLists.txt
index 7eedda0..ccf4ba3 100644
--- a/sftp/CMakeLists.txt
+++ b/sftp/CMakeLists.txt
@@ -4,7 +4,8 @@ include_directories(${LIBSSH_INCLUDE_DIR})
 
 add_library(kio_sftp MODULE kio_sftp.cpp)
 target_link_libraries(kio_sftp KF5::KIOCore ${LIBSSH_LIBRARIES} KF5::KDE4Support)
+set_target_properties(kio_sftp PROPERTIES OUTPUT_NAME "sftp")
 
-install(TARGETS kio_sftp  DESTINATION ${PLUGIN_INSTALL_DIR} )
+install(TARGETS kio_sftp  DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio )
 
 install( FILES sftp.protocol  DESTINATION  ${SERVICES_INSTALL_DIR} )
diff --git a/sftp/sftp.protocol b/sftp/sftp.protocol
index ec15eeb..6a45c08 100644
--- a/sftp/sftp.protocol
+++ b/sftp/sftp.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_sftp
+exec=kf5/kio/sftp
 protocol=sftp
 input=none
 listing=Name,Type,Size,Date,AccessDate,Access,Owner,Group,Link,MimeType
diff --git a/smb/CMakeLists.txt b/smb/CMakeLists.txt
index a575f9b..fea6edc 100644
--- a/smb/CMakeLists.txt
+++ b/smb/CMakeLists.txt
@@ -33,7 +33,9 @@ add_library(kio_smb MODULE ${kio_smb_PART_SRCS})
 target_link_libraries(kio_smb KF5::KIOCore KF5::I18n mpr)    
 endif()
 
-install(TARGETS kio_smb  DESTINATION ${PLUGIN_INSTALL_DIR} )
+set_target_properties(kio_smb PROPERTIES OUTPUT_NAME "smb")
+
+install(TARGETS kio_smb DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 
 
 ########### install files ###############
diff --git a/smb/smb.protocol b/smb/smb.protocol
index e585978..8a01964 100644
--- a/smb/smb.protocol
+++ b/smb/smb.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_smb
+exec=kf5/kio/smb
 protocol=smb
 input=none
 output=filesystem
diff --git a/thumbnail/CMakeLists.txt b/thumbnail/CMakeLists.txt
index cd1263a..c2bce99 100644
--- a/thumbnail/CMakeLists.txt
+++ b/thumbnail/CMakeLists.txt
@@ -45,7 +45,9 @@ target_link_libraries(kio_thumbnail
 	KF5::KDE4Support
 )
 
-install(TARGETS kio_thumbnail DESTINATION ${PLUGIN_INSTALL_DIR})
+set_target_properties(kio_thumbnail PROPERTIES OUTPUT_NAME "thumbnail")
+
+install(TARGETS kio_thumbnail DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 
 ########### next target ###############
 
diff --git a/thumbnail/thumbnail.protocol b/thumbnail/thumbnail.protocol
index eef743a..360e68f 100644
--- a/thumbnail/thumbnail.protocol
+++ b/thumbnail/thumbnail.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_thumbnail
+exec=kf5/kio/thumbnail
 protocol=thumbnail
 input=stream
 output=stream
