diff --git a/src/kpassworddialog.cpp b/src/kpassworddialog.cpp
index cacf31a..88c821e 100644
--- a/src/kpassworddialog.cpp
+++ b/src/kpassworddialog.cpp
@@ -401,4 +401,9 @@ bool KPasswordDialog::checkPassword()
     return true;
 }
 
+QDialogButtonBox *KPasswordDialog::buttonBox() const
+{
+    return d->ui.buttonBox;
+}
+
 #include "moc_kpassworddialog.cpp"
diff --git a/src/kpassworddialog.h b/src/kpassworddialog.h
index 069e301..e7171c7 100644
--- a/src/kpassworddialog.h
+++ b/src/kpassworddialog.h
@@ -267,6 +267,16 @@ public:
      */
     void accept();
 
+    /**
+     * Returns the button box used in the dialog.
+     * This can be used to add new buttons.
+     *
+     * @return the button box
+     *
+     * @since 5.0
+     */
+    QDialogButtonBox *buttonBox() const;
+
 Q_SIGNALS:
     /**
      * emitted when the dialog has been accepted
