diff --git a/akonadi/src/selftest/main.cpp b/akonadi/src/selftest/main.cpp
index 809dc91..6cee771 100644
--- a/akonadi/src/selftest/main.cpp
+++ b/akonadi/src/selftest/main.cpp
@@ -32,7 +32,7 @@ int main(int argc, char *argv[])
                    i18n("Akonadi Self Test"),
                    QStringLiteral("1.0"),
                    i18n("Checks and reports state of Akonadi server"),
-                   KAboutLicense::GPL_V2,
+                   KAboutData::License_GPL_V2,
                    i18n("(c) 2008 Volker Krause <vkrause@kde.org>"));
   about.setupCommandLine(&parser);
 
