diff --git a/src/kioslaves/search/CMakeLists.txt b/src/kioslaves/search/CMakeLists.txt
index 4a1bf69..2e9d4a1 100644
--- a/src/kioslaves/search/CMakeLists.txt
+++ b/src/kioslaves/search/CMakeLists.txt
@@ -6,5 +6,7 @@ target_link_libraries(kio_baloosearch
   KF5::BalooCore
 )
 
-install(TARGETS kio_baloosearch DESTINATION ${PLUGIN_INSTALL_DIR})
+set_target_properties(kio_baloosearch PROPERTIES OUTPUT_NAME baloosearch)
+
+install(TARGETS kio_baloosearch DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 install(FILES baloosearch.protocol DESTINATION ${SERVICES_INSTALL_DIR})
diff --git a/src/kioslaves/search/baloosearch.protocol b/src/kioslaves/search/baloosearch.protocol
index f241ac6..07f5fc8 100644
--- a/src/kioslaves/search/baloosearch.protocol
+++ b/src/kioslaves/search/baloosearch.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_baloosearch
+exec=baloosearch
 protocol=baloosearch
 input=none
 output=filesystem
diff --git a/src/kioslaves/tags/CMakeLists.txt b/src/kioslaves/tags/CMakeLists.txt
index f379834..00e3b05 100644
--- a/src/kioslaves/tags/CMakeLists.txt
+++ b/src/kioslaves/tags/CMakeLists.txt
@@ -1,6 +1,6 @@
-add_library(kio_tags MODULE kio_tags.cpp)
+add_library(tags MODULE kio_tags.cpp)
 
-target_link_libraries(kio_tags
+target_link_libraries(tags
   KF5::KIOWidgets
   KF5::FileMetaData
   KF5::I18n
@@ -8,5 +8,5 @@ target_link_libraries(kio_tags
   KF5::BalooFiles
 )
 
-install(TARGETS kio_tags DESTINATION ${PLUGIN_INSTALL_DIR})
+install(TARGETS tags DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 install( FILES tags.protocol DESTINATION ${SERVICES_INSTALL_DIR})
diff --git a/src/kioslaves/tags/tags.protocol b/src/kioslaves/tags/tags.protocol
index f14af6d..7a6d877 100644
--- a/src/kioslaves/tags/tags.protocol
+++ b/src/kioslaves/tags/tags.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_tags
+exec=tags
 protocol=tags
 input=none
 output=filesystem
diff --git a/src/kioslaves/timeline/CMakeLists.txt b/src/kioslaves/timeline/CMakeLists.txt
index 7c9e35c..5c23506 100644
--- a/src/kioslaves/timeline/CMakeLists.txt
+++ b/src/kioslaves/timeline/CMakeLists.txt
@@ -1,16 +1,16 @@
-set(kio_timeline_SRCS
+set(timeline_SRCS
   kio_timeline.cpp
   timelinetools.cpp
 )
 
-add_library(kio_timeline MODULE ${kio_timeline_SRCS})
+add_library(timeline MODULE ${kio_timeline_SRCS})
 
-target_link_libraries(kio_timeline
+target_link_libraries(timeline
   KF5::KIOWidgets
   KF5::I18n
   KF5::KDELibs4Support
   KF5::BalooCore
 )
 
-install(TARGETS kio_timeline DESTINATION ${PLUGIN_INSTALL_DIR})
+install(TARGETS timeline DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
 install(FILES timeline.protocol DESTINATION ${SERVICES_INSTALL_DIR})
diff --git a/src/kioslaves/timeline/timeline.protocol b/src/kioslaves/timeline/timeline.protocol
index 81f814e..bacfecb 100644
--- a/src/kioslaves/timeline/timeline.protocol
+++ b/src/kioslaves/timeline/timeline.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-exec=kio_timeline
+exec=timeline
 protocol=timeline
 input=none
 output=filesystem
