diff --git a/kate/src/main.cpp b/kate/src/main.cpp
index 1453cba..07a6e91 100644
--- a/kate/src/main.cpp
+++ b/kate/src/main.cpp
@@ -55,7 +55,7 @@ extern "C" Q_DECL_EXPORT int kdemain(int argc, char **argv)
      * construct about data for Kate
      */
     KAboutData aboutData(QStringLiteral("kate"), i18n("Kate"), QStringLiteral(KATE_VERSION),
-                         i18n("Kate - Advanced Text Editor"), KAboutLicense::LGPL_V2,
+                         i18n("Kate - Advanced Text Editor"), KAboutData::License_LGPL_V2,
                          i18n("(c) 2000-2014 The Kate Authors"), QString(), QStringLiteral("http://kate-editor.org"));
 
     /**
diff --git a/kwrite/main.cpp b/kwrite/main.cpp
index 719f045..883f403 100644
--- a/kwrite/main.cpp
+++ b/kwrite/main.cpp
@@ -48,7 +48,7 @@ extern "C" Q_DECL_EXPORT int kdemain(int argc, char **argv)
     KAboutData aboutData(QStringLiteral("kwrite"),
                          i18n("KWrite"),
                          QStringLiteral(KATE_VERSION),
-                         i18n("KWrite - Text Editor"), KAboutLicense::LGPL_V2,
+                         i18n("KWrite - Text Editor"), KAboutData::License_LGPL_V2,
                          i18n("(c) 2000-2014 The Kate Authors"), QString(), QStringLiteral("http://kate-editor.org"));
 
     /**
