diff --git a/src/ConfigureChecks.cmake b/src/ConfigureChecks.cmake
index 70b9f7d..01346e4 100644
--- a/src/ConfigureChecks.cmake
+++ b/src/ConfigureChecks.cmake
@@ -146,7 +146,9 @@ elseif(KAUTH_BACKEND_NAME STREQUAL "POLKITQT5-1")
         backends/polkit-1/Polkit1Backend.cpp
     )
 
-    set(KAUTH_BACKEND_LIBS PolkitQt5-1::Core Qt5::DBus Qt5::Widgets)
+    include_directories(${POLKITQT-1_INCLUDE_DIR})
+
+    set(KAUTH_BACKEND_LIBS polkit-qt5-core-1 Qt5::DBus Qt5::Widgets)
 
     # POLKITQT-1_POLICY_FILES_INSTALL_DIR has an absolute pathname, fix that.
     string(REPLACE ${POLKITQT-1_INSTALL_DIR}
