diff --git a/CMakeLists.txt b/CMakeLists.txt
index ccc5455..61cbbd7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -109,8 +109,8 @@ set_package_properties(OpenGLES PROPERTIES DESCRIPTION "The OpenGLES libraries"
                        TYPE OPTIONAL
                       )
 
-find_package(KActivities)
-set_package_properties(KActivities PROPERTIES DESCRIPTION "The KActivities library"
+find_package(KF5Activities)
+set_package_properties(KF5Activities PROPERTIES DESCRIPTION "The KActivities library"
                        URL "https://projects.kde.org/kactivities"
                        TYPE OPTIONAL
                       )
@@ -249,7 +249,7 @@ if(UNIX)
    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -std=c++0x")
 endif()
 
-include_directories(${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR} ${KDE4_INCLUDES} ${KACTIVITIES_INCLUDE_DIRS} ${CMAKE_SOURCE_DIR}/libs)
+include_directories(${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR} ${KDE4_INCLUDES} ${CMAKE_SOURCE_DIR}/libs)
 
 # libs
 # add_subdirectory(cmake)
diff --git a/kwin/CMakeLists.txt b/kwin/CMakeLists.txt
index 4fbcdbe..29bac16 100644
--- a/kwin/CMakeLists.txt
+++ b/kwin/CMakeLists.txt
@@ -11,7 +11,7 @@ option(KWIN_BUILD_SCRIPTING "Enable building of KWin with scripting support" ON)
 option(KWIN_BUILD_KAPPMENU "Enable building of KWin with application menu support" ON)
 option(KWIN_BUILD_XRENDER_COMPOSITING "Enable building of KWin with XRender Compositing support" ON)
 option(KWIN_BUILD_OPENGL_1_COMPOSITING "Enable support for OpenGL 1.x, automatically disabled when building for OpenGL ES 2.0" ON)
-cmake_dependent_option(KWIN_BUILD_ACTIVITIES "Enable building of KWin with kactivities support" ON "KActivities_FOUND" OFF)
+cmake_dependent_option(KWIN_BUILD_ACTIVITIES "Enable building of KWin with kactivities support" ON "KF5Activities_FOUND" OFF)
 option(KWIN_PLASMA_ACTIVE "Enable building KWin for Plasma Active." OFF)
 
 # Binary name of KWin
@@ -308,7 +308,7 @@ if(KWIN_BUILD_SCRIPTING)
 endif()
 
 if(KWIN_BUILD_ACTIVITIES)
-    set(kwin_KDE_LIBS ${kwin_KDE_LIBS} KF5::KActivities)
+    set(kwin_KDE_LIBS ${kwin_KDE_LIBS} KF5::Activities)
 endif()
 
 if(OPENGL_EGL_FOUND)
diff --git a/kwin/kcmkwin/kwinrules/CMakeLists.txt b/kwin/kcmkwin/kwinrules/CMakeLists.txt
index 9c63f82..485b7c0 100644
--- a/kwin/kcmkwin/kwinrules/CMakeLists.txt
+++ b/kwin/kcmkwin/kwinrules/CMakeLists.txt
@@ -29,7 +29,7 @@ set(kcm_libs
 )
 
 if(KWIN_BUILD_ACTIVITIES)
-    set(kcm_libs ${kcm_libs} KF5::KActivities)
+    set(kcm_libs ${kcm_libs} KF5::Activities)
 endif()
 
 target_link_libraries(kdeinit_kwin_rules_dialog ${kcm_libs} ${kwin_kcm_rules_XCB_LIBS})
