diff --git a/CMakeLists.txt b/CMakeLists.txt
index cef6a8e..a439dc8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -80,7 +80,7 @@ install(FILES
     polkitqt1-export.h
 
     DESTINATION
-    ${INCLUDE_INSTALL_DIR}/polkit-qt-1 COMPONENT Devel)
+    ${INCLUDE_INSTALL_DIR}/polkit-qt5-1 COMPONENT Devel)
 
 install(FILES
     includes/PolkitQt1/Authority
@@ -90,50 +90,50 @@ install(FILES
     includes/PolkitQt1/TemporaryAuthorization
     includes/PolkitQt1/ActionDescription
     DESTINATION
-    ${INCLUDE_INSTALL_DIR}/polkit-qt-1/PolkitQt1 COMPONENT Devel)
+    ${INCLUDE_INSTALL_DIR}/polkit-qt5-1/PolkitQt1 COMPONENT Devel)
 
 install(FILES
     includes/PolkitQt1/Gui/Action
     includes/PolkitQt1/Gui/ActionButton
     includes/PolkitQt1/Gui/ActionButtons
     DESTINATION
-    ${INCLUDE_INSTALL_DIR}/polkit-qt-1/PolkitQt1/Gui COMPONENT Devel)
+    ${INCLUDE_INSTALL_DIR}/polkit-qt5-1/PolkitQt1/Gui COMPONENT Devel)
 
 install(FILES
     includes/PolkitQt1/Agent/Listener
     includes/PolkitQt1/Agent/Session
     DESTINATION
-    ${INCLUDE_INSTALL_DIR}/polkit-qt-1/PolkitQt1/Agent COMPONENT Devel)
+    ${INCLUDE_INSTALL_DIR}/polkit-qt5-1/PolkitQt1/Agent COMPONENT Devel)
 
 if(NOT WIN32)
     # Pkgconfig
-    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/polkit-qt-1.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt-1.pc @ONLY)
-    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt-1.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
-    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/polkit-qt-core-1.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt-core-1.pc
+    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/polkit-qt-1.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt5-1.pc @ONLY)
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt5-1.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
+    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/polkit-qt-core-1.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt5-core-1.pc
                    @ONLY)
-    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt-core-1.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
-    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/polkit-qt-gui-1.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt-gui-1.pc @ONLY)
-    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt-gui-1.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
-    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/polkit-qt-agent-1.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt-agent-1.pc
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt5-core-1.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
+    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/polkit-qt-gui-1.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt5-gui-1.pc @ONLY)
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt5-gui-1.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
+    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/polkit-qt-agent-1.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt5-agent-1.pc
                    @ONLY)
-    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt-agent-1.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt5-agent-1.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
 endif(NOT WIN32)
 
 # CMake Config files
-configure_file(PolkitQt-1Config.cmake.in "${CMAKE_BINARY_DIR}/PolkitQt-1Config.cmake" @ONLY)
+configure_file(PolkitQt-1Config.cmake.in "${CMAKE_BINARY_DIR}/PolkitQt5-1Config.cmake" @ONLY)
 
 # this file is used by to check if the installed version can be used.
-macro_write_basic_cmake_version_file(${CMAKE_BINARY_DIR}/PolkitQt-1ConfigVersion.cmake
+macro_write_basic_cmake_version_file(${CMAKE_BINARY_DIR}/PolkitQt5-1ConfigVersion.cmake
                                      ${POLKITQT-1_VERSION_MAJOR} ${POLKITQT-1_VERSION_MINOR} ${POLKITQT-1_VERSION_PATCH})
 
 if(USE_COMMON_CMAKE_PACKAGE_CONFIG_DIR)
-   set(_PolkitQt-1Config_INSTALL_DIR ${LIB_INSTALL_DIR}/cmake/PolkitQt-1)
+   set(_PolkitQt-1Config_INSTALL_DIR ${LIB_INSTALL_DIR}/cmake/PolkitQt5-1)
 else(USE_COMMON_CMAKE_PACKAGE_CONFIG_DIR)
-   set(_PolkitQt-1Config_INSTALL_DIR ${LIB_INSTALL_DIR}/PolkitQt-1/cmake)
+   set(_PolkitQt-1Config_INSTALL_DIR ${LIB_INSTALL_DIR}/PolkitQt5-1/cmake)
 endif(USE_COMMON_CMAKE_PACKAGE_CONFIG_DIR)
 
-install(FILES ${CMAKE_BINARY_DIR}/PolkitQt-1ConfigVersion.cmake
-              ${CMAKE_BINARY_DIR}/PolkitQt-1Config.cmake
+install(FILES ${CMAKE_BINARY_DIR}/PolkitQt5-1ConfigVersion.cmake
+              ${CMAKE_BINARY_DIR}/PolkitQt5-1Config.cmake
         DESTINATION ${_PolkitQt-1Config_INSTALL_DIR} )
 
 option(BUILD_EXAMPLES "Builds a set of examples for polkit-qt-1" OFF)
diff --git a/PolkitQt-1Config.cmake.in b/PolkitQt-1Config.cmake.in
index db60830..d3d2812 100644
--- a/PolkitQt-1Config.cmake.in
+++ b/PolkitQt-1Config.cmake.in
@@ -13,29 +13,29 @@ if(NOT POLKITQT-1_INSTALL_DIR)
    set(POLKITQT-1_INSTALL_DIR "@CMAKE_INSTALL_PREFIX@")
 endif(NOT POLKITQT-1_INSTALL_DIR)
 
-set(POLKITQT-1_INCLUDE_DIR              "${POLKITQT-1_INSTALL_DIR}/@INCLUDE_INSTALL_DIR@/polkit-qt-1")
+set(POLKITQT-1_INCLUDE_DIR              "${POLKITQT-1_INSTALL_DIR}/@INCLUDE_INSTALL_DIR@/polkit-qt5-1")
 set(POLKITQT-1_LIB_DIR                  "${POLKITQT-1_INSTALL_DIR}/@LIB_INSTALL_DIR@")
 set(POLKITQT-1_POLICY_FILES_INSTALL_DIR "${POLKITQT-1_INSTALL_DIR}/share/polkit-1/actions")
 
 # Compatibility
 if(WIN32)
 if(MINGW)
-  set(POLKITQT-1_CORE_LIBRARY         "${POLKITQT-1_LIB_DIR}/libpolkit-qt-core-1.dll.a")
-  set(POLKITQT-1_AGENT_LIBRARY        "${POLKITQT-1_LIB_DIR}/libpolkit-qt-agent-1.dll.a")
-  set(POLKITQT-1_GUI_LIBRARY          "${POLKITQT-1_LIB_DIR}/libpolkit-qt-gui-1.dll.a")
+  set(POLKITQT-1_CORE_LIBRARY         "${POLKITQT-1_LIB_DIR}/libpolkit-qt5-core-1.dll.a")
+  set(POLKITQT-1_AGENT_LIBRARY        "${POLKITQT-1_LIB_DIR}/libpolkit-qt5-agent-1.dll.a")
+  set(POLKITQT-1_GUI_LIBRARY          "${POLKITQT-1_LIB_DIR}/libpolkit-qt5-gui-1.dll.a")
 else(MINGW)
-  set(POLKITQT-1_CORE_LIBRARY         "${POLKITQT-1_LIB_DIR}/polkit-qt-core-1.lib")
-  set(POLKITQT-1_AGENT_LIBRARY        "${POLKITQT-1_LIB_DIR}/polkit-qt-agent-1.lib")
-  set(POLKITQT-1_GUI_LIBRARY          "${POLKITQT-1_LIB_DIR}/polkit-qt-gui-1.lib")
+  set(POLKITQT-1_CORE_LIBRARY         "${POLKITQT-1_LIB_DIR}/polkit-qt5-core-1.lib")
+  set(POLKITQT-1_AGENT_LIBRARY        "${POLKITQT-1_LIB_DIR}/polkit-qt5-agent-1.lib")
+  set(POLKITQT-1_GUI_LIBRARY          "${POLKITQT-1_LIB_DIR}/polkit-qt5-gui-1.lib")
 endif(MINGW)
 elseif(APPLE)
-  set(POLKITQT-1_CORE_LIBRARY         "${POLKITQT-1_LIB_DIR}/libpolkit-qt-core-1.dylib")
-  set(POLKITQT-1_AGENT_LIBRARY        "${POLKITQT-1_LIB_DIR}/libpolkit-qt-agent-1.dylib")
-  set(POLKITQT-1_GUI_LIBRARY          "${POLKITQT-1_LIB_DIR}/libpolkit-qt-gui-1.dylib")
+  set(POLKITQT-1_CORE_LIBRARY         "${POLKITQT-1_LIB_DIR}/libpolkit-qt5-core-1.dylib")
+  set(POLKITQT-1_AGENT_LIBRARY        "${POLKITQT-1_LIB_DIR}/libpolkit-qt5-agent-1.dylib")
+  set(POLKITQT-1_GUI_LIBRARY          "${POLKITQT-1_LIB_DIR}/libpolkit-qt5-gui-1.dylib")
 else()
-  set(POLKITQT-1_CORE_LIBRARY         "${POLKITQT-1_LIB_DIR}/libpolkit-qt-core-1.so")
-  set(POLKITQT-1_AGENT_LIBRARY        "${POLKITQT-1_LIB_DIR}/libpolkit-qt-agent-1.so")
-  set(POLKITQT-1_GUI_LIBRARY          "${POLKITQT-1_LIB_DIR}/libpolkit-qt-gui-1.so")
+  set(POLKITQT-1_CORE_LIBRARY         "${POLKITQT-1_LIB_DIR}/libpolkit-qt5-core-1.so")
+  set(POLKITQT-1_AGENT_LIBRARY        "${POLKITQT-1_LIB_DIR}/libpolkit-qt5-agent-1.so")
+  set(POLKITQT-1_GUI_LIBRARY          "${POLKITQT-1_LIB_DIR}/libpolkit-qt5-gui-1.so")
 endif()
 
 set(POLKITQT-1_LIBRARIES ${POLKITQT-1_GUI_LIBRARY} ${POLKITQT-1_CORE_LIBRARY} ${POLKITQT-1_AGENT_LIBRARY})
diff --git a/agent/CMakeLists.txt b/agent/CMakeLists.txt
index 4df7d62..b0cc05b 100644
--- a/agent/CMakeLists.txt
+++ b/agent/CMakeLists.txt
@@ -9,17 +9,17 @@ set(polkit_qt_agent_SRCS
     listeneradapter.cpp
     polkitqtlistener.cpp
 )
-add_library(polkit-qt-agent-1 SHARED ${polkit_qt_agent_SRCS})
+add_library(polkit-qt5-agent-1 SHARED ${polkit_qt_agent_SRCS})
 
-target_link_libraries(polkit-qt-agent-1
+target_link_libraries(polkit-qt5-agent-1
     ${POLKIT_LIBRARIES}
     ${QT_QTCORE_LIBRARY}
     ${POLKIT_AGENT_LIBRARY}
-    polkit-qt-core-1
+    polkit-qt5-core-1
 )
 
-set_target_properties(polkit-qt-agent-1 PROPERTIES VERSION ${POLKITQT-1_LIBRARY_VERSION}
+set_target_properties(polkit-qt5-agent-1 PROPERTIES VERSION ${POLKITQT-1_LIBRARY_VERSION}
                                 SOVERSION ${POLKITQT-1_ABI_VERSION}
                                 DEFINE_SYMBOL MAKE_POLKITQT1_LIB)
 
-install(TARGETS polkit-qt-agent-1 ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS polkit-qt5-agent-1 ${INSTALL_TARGETS_DEFAULT_ARGS})
diff --git a/core/CMakeLists.txt b/core/CMakeLists.txt
index 5afdae4..8dbd258 100644
--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -12,21 +12,21 @@ set(polkit_qt_core_SRCS
     polkitqt1-actiondescription.cpp
 )
 
-add_library(polkit-qt-core-1 SHARED ${polkit_qt_core_SRCS})
+add_library(polkit-qt5-core-1 SHARED ${polkit_qt_core_SRCS})
 
-target_link_libraries(polkit-qt-core-1
+target_link_libraries(polkit-qt5-core-1
     ${POLKIT_LIBRARIES}
     ${GLIB2_LIBRARIES}
     ${GOBJECT_LIBRARIES}
     ${GIO_LIBRARIES}
 )
 
-qt5_use_modules(polkit-qt-core-1
+qt5_use_modules(polkit-qt5-core-1
     Xml DBus
 )
 
-set_target_properties(polkit-qt-core-1 PROPERTIES VERSION ${POLKITQT-1_LIBRARY_VERSION}
+set_target_properties(polkit-qt5-core-1 PROPERTIES VERSION ${POLKITQT-1_LIBRARY_VERSION}
                                 SOVERSION ${POLKITQT-1_ABI_VERSION}
                                 DEFINE_SYMBOL MAKE_POLKITQT1_LIB)
 
-install(TARGETS polkit-qt-core-1 ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS polkit-qt5-core-1 ${INSTALL_TARGETS_DEFAULT_ARGS})
diff --git a/gui/CMakeLists.txt b/gui/CMakeLists.txt
index 47ac74d..4c4b389 100644
--- a/gui/CMakeLists.txt
+++ b/gui/CMakeLists.txt
@@ -4,18 +4,18 @@ set(polkit_qt_gui_SRCS
     polkitqt1-gui-actionbuttons.cpp
 )
 
-add_library(polkit-qt-gui-1 SHARED ${polkit_qt_gui_SRCS})
+add_library(polkit-qt5-gui-1 SHARED ${polkit_qt_gui_SRCS})
 
-target_link_libraries(polkit-qt-gui-1
+target_link_libraries(polkit-qt5-gui-1
     ${POLKIT_LIBRARIES}
-    polkit-qt-core-1
+    polkit-qt5-core-1
 )
-qt5_use_modules(polkit-qt-gui-1
+qt5_use_modules(polkit-qt5-gui-1
     Widgets DBus
 )
 
-set_target_properties(polkit-qt-gui-1 PROPERTIES VERSION ${POLKITQT-1_LIBRARY_VERSION}
+set_target_properties(polkit-qt5-gui-1 PROPERTIES VERSION ${POLKITQT-1_LIBRARY_VERSION}
                                 SOVERSION ${POLKITQT-1_ABI_VERSION}
                                 DEFINE_SYMBOL MAKE_POLKITQT1_LIB)
 
-install(TARGETS polkit-qt-gui-1 ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS polkit-qt5-gui-1 ${INSTALL_TARGETS_DEFAULT_ARGS})
diff --git a/polkit-qt-1.pc.cmake b/polkit-qt-1.pc.cmake
index 0d7bd08..fa4a637 100644
--- a/polkit-qt-1.pc.cmake
+++ b/polkit-qt-1.pc.cmake
@@ -3,9 +3,9 @@ exec_prefix=@CMAKE_INSTALL_PREFIX@
 libdir=@LIB_INSTALL_DIR@
 includedir=@CMAKE_INSTALL_PREFIX@/include
 
-Name: polkit-qt-1
+Name: polkit-qt5-1
 Description: Convenience library for using polkit with a Qt-styled API
 Version: @POLKITQT-1_VERSION_STRING@
-Requires: polkit-qt-core-1 polkit-qt-gui-1 polkit-qt-agent-1
-Libs: -L${libdir} -lpolkit-qt-core-1 -lpolkit-qt-gui-1 -lpolkit-qt-agent-1
+Requires: polkit-qt5-core-1 polkit-qt5-gui-1 polkit-qt5-agent-1
+Libs: -L${libdir} -lpolkit-qt5-core-1 -lpolkit-qt5-gui-1 -lpolkit-qt5-agent-1
 Cflags: -I${includedir}
diff --git a/polkit-qt-agent-1.pc.cmake b/polkit-qt-agent-1.pc.cmake
index 09f9323..7edbcde 100644
--- a/polkit-qt-agent-1.pc.cmake
+++ b/polkit-qt-agent-1.pc.cmake
@@ -3,9 +3,9 @@ exec_prefix=@CMAKE_INSTALL_PREFIX@
 libdir=@LIB_INSTALL_DIR@
 includedir=@CMAKE_INSTALL_PREFIX@/include
 
-Name: polkit-qt-agent-1
+Name: polkit-qt5-agent-1
 Description: Convenience library for using polkit Agent with a Qt-styled API
 Version: @POLKITQT-1_VERSION_STRING@
 Requires: QtCore QtGui
-Libs: -L${libdir} -lpolkit-qt-agent-1
+Libs: -L${libdir} -lpolkit-qt5-agent-1
 Cflags: -I${includedir}
diff --git a/polkit-qt-core-1.pc.cmake b/polkit-qt-core-1.pc.cmake
index f553b7b..b7acbfa 100644
--- a/polkit-qt-core-1.pc.cmake
+++ b/polkit-qt-core-1.pc.cmake
@@ -3,9 +3,9 @@ exec_prefix=@CMAKE_INSTALL_PREFIX@
 libdir=@LIB_INSTALL_DIR@
 includedir=@CMAKE_INSTALL_PREFIX@/include
 
-Name: polkit-qt-core-1
+Name: polkit-qt5-core-1
 Description: Convenience library for using polkit with a Qt-styled API, non-GUI classes
 Version: @POLKITQT-1_VERSION_STRING@
 Requires: QtCore
-Libs: -L${libdir} -lpolkit-qt-core-1
+Libs: -L${libdir} -lpolkit-qt5-core-1
 Cflags: -I${includedir}
diff --git a/polkit-qt-gui-1.pc.cmake b/polkit-qt-gui-1.pc.cmake
index 83d4e9a..d8ea7e5 100644
--- a/polkit-qt-gui-1.pc.cmake
+++ b/polkit-qt-gui-1.pc.cmake
@@ -3,9 +3,9 @@ exec_prefix=@CMAKE_INSTALL_PREFIX@
 libdir=@LIB_INSTALL_DIR@
 includedir=@CMAKE_INSTALL_PREFIX@/include
 
-Name: polkit-qt-gui-1
+Name: polkit-qt5-gui-1
 Description: Convenience library for using polkit with a Qt-styled API, GUI classes
 Version: @POLKITQT-1_VERSION_STRING@
-Requires: QtCore QtGui polkit-qt-core-1
-Libs: -L${libdir} -lpolkit-qt-gui-1
+Requires: QtCore QtGui polkit-qt5-core-1
+Libs: -L${libdir} -lpolkit-qt5-gui-1
 Cflags: -I${includedir}
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index cdafe19..73aa18d 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -4,15 +4,15 @@ include_directories(
     ${CMAKE_SOURCE_DIR}/agent
 )
 
-add_executable(polkit-qt-test
+add_executable(polkit-qt5-test
     test.cpp
 )
 
-target_link_libraries(polkit-qt-test
-    polkit-qt-core-1
+target_link_libraries(polkit-qt5-test
+    polkit-qt5-core-1
 )
-qt5_use_modules(polkit-qt-test
+qt5_use_modules(polkit-qt5-test
     Core DBus Test
 )
 
-add_test(BaseTest ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt-test)
+add_test(BaseTest ${CMAKE_CURRENT_BINARY_DIR}/polkit-qt5-test)
