diff --git a/src/request.cpp b/src/request.cpp
index 5556c79..4f51e8e 100644
--- a/src/request.cpp
+++ b/src/request.cpp
@@ -81,9 +81,7 @@ public:
     }
 
 private Q_SLOTS:
-#if HAS_XEMBED
     void onEmbedError();
-#endif
     void onIndicatorCallFinished(QDBusPendingCallWatcher *watcher);
 
 private:
@@ -194,9 +192,9 @@ void RequestPrivate::setWidget(QWidget *widget)
 #endif
 }
 
-#if HAS_XEMBED
 void RequestPrivate::onEmbedError()
 {
+#if HAS_XEMBED
     Q_Q(Request);
 
     QX11EmbedWidget *embed = qobject_cast<QX11EmbedWidget*>(sender());
@@ -204,8 +202,8 @@ void RequestPrivate::onEmbedError()
 
     q->fail(SIGNON_UI_ERROR_EMBEDDING_FAILED,
             QString("Embedding signon UI failed: %1").arg(embed->error()));
-}
 #endif
+}
 
 Accounts::Account *RequestPrivate::findAccount()
 {
