commit 91fffc36da8db117deea4c7c68f4b1c01523b519
Author: Kevin Funk <kfunk@kde.org>
Date:   Mon May 5 01:42:43 2014 +0200

    Improve error reporting in meinproc

diff --git a/src/xslt.cpp b/src/xslt.cpp
index 34166ae..d98c341 100644
--- a/src/xslt.cpp
+++ b/src/xslt.cpp
@@ -398,10 +398,11 @@ QStringList locateFilesInDtdResource(const QString &file, const QStandardPaths::
     // prefix.
     // This is the case on build.kde.org where kdelibs4support installs catalogs
     // in a different prefix than kdoctools.
+    const QString fileName = QStringLiteral("kf5/kdoctools/") + file;
     const QStringList result = QStandardPaths::locateAll(QStandardPaths::GenericDataLocation,
-            QStringLiteral("kf5/kdoctools/") + file, option);
+            fileName, option);
     if (result.isEmpty()) {
-        qDebug() << "Could not locate file" << file << "in standard dirs";
+        qDebug() << "Could not locate file" << fileName << "in" << QStandardPaths::standardLocations(QStandardPaths::GenericDataLocation);
     }
     return result;
 }
