diff --git a/plasma-workspace/startkde/startkde.cmake b/plasma-workspace/startkde/startkde.cmake
index e845603..a742dcb 100644
--- a/plasma-workspace/startkde/startkde.cmake
+++ b/plasma-workspace/startkde/startkde.cmake
@@ -332,7 +332,7 @@ export XDG_CURRENT_DESKTOP
 
 # We set LD_BIND_NOW to increase the efficiency of kdeinit.
 # kdeinit unsets this variable before loading applications.
-LD_BIND_NOW=true @CMAKE_INSTALL_PREFIX@/@LIBEXEC_INSTALL_DIR@/start_kdeinit_wrapper --kded +kcminit_startup
+LD_BIND_NOW=true /usr/libexec/start_kdeinit_wrapper --kded +kcminit_startup
 if test $? -ne 0; then
   # Startup error
   echo 'startkde: Could not start kdeinit5. Check your installation.'  1>&2
