diff --git a/src/start_kdeinit/start_kdeinit.c b/src/start_kdeinit/start_kdeinit.c
index 02d4d54..8ddef1f 100644
--- a/src/start_kdeinit/start_kdeinit.c
+++ b/src/start_kdeinit/start_kdeinit.c
@@ -32,7 +32,7 @@
 #include <sys/capability.h>
 #endif
 
-#define EXECUTE CMAKE_INSTALL_PREFIX"/"BIN_INSTALL_DIR "/kdeinit5"
+#define EXECUTE BIN_INSTALL_DIR "/kdeinit5"
 
 #if KDEINIT_OOM_PROTECT
 
diff --git a/src/start_kdeinit/start_kdeinit_wrapper.c b/src/start_kdeinit/start_kdeinit_wrapper.c
index f35c4a9..9fdb6a0 100644
--- a/src/start_kdeinit/start_kdeinit_wrapper.c
+++ b/src/start_kdeinit/start_kdeinit_wrapper.c
@@ -23,7 +23,7 @@
 #include <string.h>
 #include <unistd.h>
 
-#define EXECUTE CMAKE_INSTALL_PREFIX"/"KF5_LIBEXEC_INSTALL_DIR "/start_kdeinit"
+#define EXECUTE KF5_LIBEXEC_INSTALL_DIR "/start_kdeinit"
 
 #if KDEINIT_OOM_PROTECT
 
